
<!doctype html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="google-site-verification" content="sKCEBeALSs13uui0cWMviq9hlm0rPByKQKNHnOdq-d0" />
    <meta name="baidu-site-verification" content="code-MBwKQgVpns" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="author" content="Songtao Gui">

    
    <meta name="description" content="Brief usage of some evolution related tools:
- jModeltest: for model selection
- phyML: ml tree
- RAxML: faster ml tree
- mrbayes: bayes tree
-paml-codeml: branch-site model
">
    

    
    
    <meta name="keywords" content="evolution_related_tools">
    <title>evolution_related_tools</title>
    

    <link rel="icon" href="/assets/favicon.svg">

    <!-- Dark mode -->
    <!-- <script src="https://cdn.jsdelivr.net/npm/darkmode-js@1.5.7/lib/darkmode-js.min.js"></script>
    <script>
        function addDarkmodeWidget()
        {
            const options = {
                top: 'unset',
                bottom: 'unset', /* default: '32px' */
                right: 'unset', /* default: '32px' */
                left: '50px', /* default: 'unset' */
                time: '.0s', /* default: '0.3s' */
                mixColor: '#fff', /* default: '#fff' */
                backgroundColor: '#fff', /* default: '#fff' */
                buttonColorDark: '#100f2c', /* default: '#100f2c' */
                buttonColorLight: '#fff', /* default: '#fff' */
                saveInCookies: true, /* default: true, */
                label: 'ğŸŒ“', /* default: '' */
                autoMatchOsTheme: true /* default: true */
            };
            new Darkmode(options).showWidget();
        };
        window.addEventListener('DOMContentLoaded', addDarkmodeWidget);
    </script> -->
    
<link rel=preconnect href="https://fonts.gstatic.com" crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.1/css/all.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css" crossorigin=anonymous>
<link rel=stylesheet
    href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap"
    crossorigin=anonymous>
<!-- locally served -->
<link rel="stylesheet" href="/css/pure.css">
<link rel=stylesheet href="/libs/academic/academic.min.css">
<link rel="stylesheet" href="/css/layout.css">
<link rel="stylesheet" href="/css/extra.css">
<link rel=stylesheet href="/css/codeset.css">
<link rel=stylesheet href="/css/admonition.css">
<link rel=stylesheet href="/css/cv.css">
<link rel=stylesheet href="/css/search-box.css">
<link rel="stylesheet" href="https://cdn.staticfile.org/font-awesome/4.7.0/css/font-awesome.css">
<link rel=stylesheet href="/css/vega.css">
<!-- <link rel=stylesheet href="/css/font-awesome.css"> -->

<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>

<!-- nutshell + highlight + katex + plotly -->

<link rel=stylesheet href="/libs/highlight/styles/atom-one-light.min.css">






    <!-- Vega -->
    <script src="https://cdn.jsdelivr.net/npm/vega@5"></script>
    <script src="https://cdn.jsdelivr.net/npm/vega-lite@5"></script>
    <script src="https://cdn.jsdelivr.net/npm/vega-embed@6"></script>
    <script src="https://cdn.jsdelivr.net/npm/@svgdotjs/svg.js@3.0/dist/svg.min.js"></script>
    <style media="screen">
        /* Add space between Vega-Embed links  */
        .vega-actions a {
          margin-right: 5px;
        }
    </style>
    <!-- end Vega -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js"
        integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script>
</head>

<body id=top data-spy="scroll" data-offset="70" data-target=#navbar-main>
    <!-- Swiftype -->
    <script type="text/javascript">
        (function(w,d,t,u,n,s,e){w['SwiftypeObject']=n;w[n]=w[n]||function(){
        (w[n].q=w[n].q||[]).push(arguments);};s=d.createElement(t);
        e=d.getElementsByTagName(t)[0];s.async=1;s.src=u;e.parentNode.insertBefore(s,e);
        })(window,document,'script','//s.swiftypecdn.com/install/v2/st.js','_st');
        
        _st('install','J57vdqtTa8oYL-ikpKNG','2.0.0');
    </script>
    
     <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main>
    <div class=container>
        <!-- Main name -->
        <div class="d-none d-lg-inline-flex">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Button for narrow mode -->
        <button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar
            aria-expanded=false aria-label="Toggle navigation">
            <span><i class="fas fa-bars"></i></span>
        </button>
        <!-- Main name for mobile mode -->
        <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Menu items -->
        <div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content>
            <ul class="navbar-nav d-md-inline-flex">
                <li class=nav-item><a class=nav-link href="/posts/"><span>Posts</span></a></li>
                <li class=nav-item><a class=nav-link href="/courses/"><span>Courses</span></a></li>
                <li class=nav-item><a class=nav-link href="/notebooks/"><span>Notebooks</span></a></li>
                <li class=nav-item><a class=nav-link href="/cv/"><span>CurriculumVitae</span></a></li>
                <li class=nav-item><a class=nav-link href="/tag/"><span>Tags</span></a></li>
            </ul>
        </div>
        <!-- <div class="nav-linksmall d-none d-lg-inline-flex inputBox">
                <input type="text" class="st-default-search-input" placeholder="Input then press ENTER" value="" required>
                <div class="search"></div>
            <script>
                var searchForm = document.getElementById('searchForm');
                searchForm.onsubmit = function () {
                    var url = this.action;
                    var q = this.children['q'].value;
                    var q = 'site:songtaogui.github.io+' + q
                    // var site = this.children['site'].value;
                    var url = url + '?q=' + q;
                    window.open(url,'__blank');
                    return false;
                };
            </script>
        </div> -->
        <!-- åŸºäºbing siteæœç´¢: -->
        <div class="nav-linksmall d-none d-lg-inline-flex inputBox">
            <form action="https://bing.com/search" id="searchForm">
                <input type="text" name="q" value="" placeholder="Input then press ENTER" value="" required>
                <div class="search"></div>
            </form>
            <script>
                var searchForm = document.getElementById('searchForm');
                searchForm.onsubmit = function () {
                    var url = this.action;
                    var q = this.children['q'].value;
                    var q = 'site:songtaogui.github.io+' + q
                    // var site = this.children['site'].value;
                    var url = url + '?q=' + q;
                    window.open(url,'__blank');
                    return false;
                };
            </script>
        </div>
        <!-- <div class="nav-icon d-none d-lg-inline-flex"> -->
        <div class="nav-icon">
            <a class=nav-linksmall href="/giants" target="_blank"><span><img class="nav-img"
                src="/assets/img/giant.svg"></span></a>
            <a class=nav-linksmall href="/achievements/" target="_blank"><span><img class="nav-img"
                        src="/assets/img/achievement.svg"></span></a>
            <a class=nav-linksmall href="/feed.xml" target="_blank"><span><img class="nav-img"
                        src="/assets/img/rss.svg"></span></a>
            <a class=nav-linksmall href="/musics" target="_blank"><span><img class="nav-img"
                        src="/assets/img/music.svg"></span></a>
        </div>

    </div>
</nav> 

    <span class="js-widget-page d-none"></span>

    
    <article class=article>
        <div class="article-container pt-3">
             <h1>evolution_related_tools</h1> 
        </div>
        
        <div class="article-container">
            <!-- å…³é—­article-style, ä¸ç„¶è·Ÿnutshellä¸å…¼å®¹ -->
            <!-- <div class="gst-style"> -->
            <div class="layout-main">
                <!-- <div class="article-style"> -->
                
                
                <div class="franklin-content"><p> <div class="tags"><a href="/tag/" id="tag-icon"><svg width="20" height="20" viewBox="0 0 512 512"><defs><style>.cls-1{fill:#141f38}</style></defs><path class="cls-1" d="M215.8 512a76.1 76.1 0 0 1-54.17-22.44L22.44 350.37a76.59 76.59 0 0 1 0-108.32L242 22.44A76.11 76.11 0 0 1 296.2 0h139.2A76.69 76.69 0 0 1 512 76.6v139.19A76.08 76.08 0 0 1 489.56 270L270 489.56A76.09 76.09 0 0 1 215.8 512zm80.4-486.4a50.69 50.69 0 0 0-36.06 14.94l-219.6 219.6a51 51 0 0 0 0 72.13l139.19 139.19a51 51 0 0 0 72.13 0l219.6-219.61a50.67 50.67 0 0 0 14.94-36.06V76.6a51.06 51.06 0 0 0-51-51zm126.44 102.08A38.32 38.32 0 1 1 461 89.36a38.37 38.37 0 0 1-38.36 38.32zm0-51a12.72 12.72 0 1 0 12.72 12.72 12.73 12.73 0 0 0-12.72-12.76z"/><path class="cls-1" d="M217.56 422.4a44.61 44.61 0 0 1-31.76-13.16l-83-83a45 45 0 0 1 0-63.52L211.49 154a44.91 44.91 0 0 1 63.51 0l83 83a45 45 0 0 1 0 63.52L249.31 409.24a44.59 44.59 0 0 1-31.75 13.16zm-96.7-141.61a19.34 19.34 0 0 0 0 27.32l83 83a19.77 19.77 0 0 0 27.31 0l108.77-108.7a19.34 19.34 0 0 0 0-27.32l-83-83a19.77 19.77 0 0 0-27.31 0l-108.77 108.7z"/><path class="cls-1" d="M294.4 281.6a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85zM256 320a12.75 12.75 0 0 1-9.05-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2A12.8 12.8 0 0 1 256 320zM217.6 358.4a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 1 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85z"/></svg></a><a href="/tag/bioinformatics/">bioinformatics</a>, <a href="/tag/evolution/">evolution</a>&nbsp;&nbsp;&nbsp;&nbsp;<a id="tag-icon"><svg width="20" height="20" class="icon" viewBox="0 0 1024 1024" version="1.1" p-id="3825" width="64" height="64"><path d="M512 416a96 96 0 1 0 0 192 96 96 0 0 0 0-192z m511.952 102.064c-0.016-0.448-0.064-0.864-0.096-1.296a8.16 8.16 0 0 0-0.08-0.656c0-0.32-0.064-0.624-0.128-0.928-0.032-0.368-0.064-0.736-0.128-1.088-0.032-0.048-0.032-0.096-0.032-0.144a39.488 39.488 0 0 0-10.704-21.536c-32.672-39.616-71.536-74.88-111.04-107.072-85.088-69.392-182.432-127.424-289.856-150.8-62.112-13.504-124.576-14.064-187.008-2.64-56.784 10.384-111.504 32-162.72 58.784-80.176 41.92-153.392 99.696-217.184 164.48-11.808 11.984-23.552 24.224-34.288 37.248-14.288 17.328-14.288 37.872 0 55.216 32.672 39.616 71.52 74.848 111.04 107.056 85.12 69.392 182.448 127.408 289.888 150.784 62.096 13.504 124.608 14.096 187.008 2.656 56.768-10.4 111.488-32 162.736-58.768 80.176-41.936 153.376-99.696 217.184-164.48 11.792-12 23.536-24.224 34.288-37.248 5.712-5.872 9.456-13.44 10.704-21.568l0.032-0.128a12.592 12.592 0 0 0 0.128-1.088c0.064-0.304 0.096-0.624 0.128-0.928l0.08-0.656 0.096-1.28c0.032-0.656 0.048-1.296 0.048-1.952l-0.096-1.968zM512 704c-106.032 0-192-85.952-192-192s85.952-192 192-192 192 85.968 192 192c0 106.048-85.968 192-192 192z" p-id="3826"></path></svg></a><span id="busuanzi_value_page_pv"></span></div></p>
<h2 id="jmodeltest"><a href="#jmodeltest" class="header-anchor">jModeltest</a></h2>
<h3 id="installation"><a href="#installation" class="header-anchor">Installation</a></h3>
<p><a href="https://drive.google.com/drive/folders/0ByrkKOPtF_n_OUs3d0dNcnJPYXM">HERE</a></p>
<h3 id="usage"><a href="#usage" class="header-anchor">USAGE</a></h3>
<p>å¿«é€Ÿå¼€å§‹:</p>
<pre><code class="language-sh">java -jar jModelTest.jar -g 4 -i -f -AIC -BIC -a -d &lt;IN.fas_aln&gt; -o &lt;OUT&gt; 
java -jar jModelTest.jar -s 11 -g 4 -i -f -AIC -BIC -AICc -DT -p -w -a -d &lt;IN.fas_aln&gt; -o &lt;OUT&gt;</code></pre>
<p>è¯¦ç»† CMD æŒ‡ä»¤:</p>
<pre><code class="language-julia">jModelTest command usage
java -jar jModelTest.jar -d sequenceFileName
                        &#91;-getPhylip&#93;
                        &#91;-ckp checkpointFileName.ckp&#93;
                        &#91;-n executionName&#93;
                        &#91;-t fixed|BIONJ|ML&#93; &#91;-u userTreeFileName&#93; &#91;-o outputFileName&#93;
                        &#91;-S NNI|SPR|BEST&#93;
                        &#91;-AIC&#93; &#91;-AICc&#93; &#91;-BIC&#93; &#91;-DT&#93; &#91;-c confidenceInterval&#93;
                        &#91;-s 3|5|7|11|203&#93;
                        &#91;-f&#93; &#91;-i&#93; &#91;-g numberOfCategories&#93;
                        &#91;-uLNL&#93;
                        &#91;-dLRT&#93; &#91;-h confidenceInterval&#93; &#91;-hLRT&#93; &#91;-O &#123;ftvwxgp&#125;&#93;
                        &#91;-a&#93; &#91;-z&#93; &#91;-p&#93; &#91;-v&#93; &#91;-w&#93;
                        &#91;-tr numberOfThreads&#93; &#91;-machinesfile machinesFileName&#93;

     -a
         estimate model-averaged phylogeny for each active criterion &#40;e.g., -a&#41; &#40;default is false&#41;

     -AIC
         calculate the Akaike Information Criterion &#40;e.g., -AIC&#41; &#40;default is false&#41;

     -AICc
         calculate the corrected Akaike Information Criterion &#40;e.g., -AICc&#41; &#40;default is false&#41;

     -BIC
         calculate the Bayesian Information Criterion &#40;e.g., -BIC&#41; &#40;default is false&#41;

     -DT
         calculate the decision theory criterion &#40;e.g., -DT&#41; &#40;default is false&#41;

     -c confidenceInterval
         confidence interval &#40;e.g., -c 90&#41; &#40;default is 100&#41;

     -ckp checkpointFileName
         Loads a checkpointing file

     -d sequenceFileName
         input data file &#40;e.g., -d data.phy&#41;

     -dLRT
         do dynamical likelihood ratio tests &#40;e.g., -dLRT&#41;&#40;default is false&#41;

     -f
         include models with unequals base frecuencies &#40;e.g., -f&#41; &#40;default is false&#41;

     -g numberOfCategories
         include models with rate variation among sites and number of categories &#40;e.g., -g 8&#41; &#40;default is false &amp; 4 categories&#41;

     -G threshold
         heuristic search. Requires a threshold &gt; 0 &#40;e.g., -G 0.1&#41;

     -getPhylip
         converts the input file into phylip format

     -h confidenceInterval
         confidence level for the hLRTs &#40;e.g., -a0.002&#41; &#40;default is 0.01&#41;

     -H informationCriterion
         information criterion for clustering search &#40;AIC, AICc, BIC&#41;. &#40;default is BIC&#41;
         this argument applies only for 203 substitution schemes &#40;e.g., -s 203 -H AIC&#41;

     -help
         displays this help message

     -hLRT
         do hierarchical likelihood ratio tests &#40;default is false&#41;
         hypothesis testing order can be specified with -O argument

     -i
         include models with a proportion invariable sites &#40;e.g., -i&#41; &#40;default is false&#41;

     -machinesfile manchinesFileName
         gets the processors per host from a machines file

     -n executionName
         execution name for appending to the log filenames &#40;default: current time yyyyMMddhhmmss&#41;

     -o outputFileName
         set output file &#40;e.g., -o jmodeltest.out&#41;

     -O hypothesisOrder
         hypothesis order for the hLRTs &#40;e.g., -hLRT -O gpftv&#41; &#40;default is ftvwxgp&#41;
            f&#61;freq, t&#61;titvi, v&#61;2ti4tv&#40;subst&#61;3&#41;/2ti&#40;subst&gt;3&#41;, w&#61;2tv, x&#61;4tv, g&#61;gamma, p&#61;pinv
            this argument is used only if -hLRT argument is set
            &#39;f&#39;,&#39;t&#39;,&#39;v&#39;,&#39;g&#39;,&#39;p&#39; are mandatory in any order. &#39;w&#39; is optional, and &#39;x&#39; requires &#39;w&#39; to be present
            thus, length should be 5, 6 *including &#39;w&#39;&#41; or 7 &#40;including both &#39;w&#39; and &#39;x&#39;&#41;
            e.g., -hLRT -O gpfvwxt

     -p
         calculate parameter importances &#40;e.g., -p&#41; &#40;default is false&#41;

     -r
         backward selection for the hLRT &#40;e.g., -r&#41; &#40;default is forward&#41;

     -s numberOfSubstitutionSchemes
         number of substitution schemes &#40;e.g., -s 11&#41; &#40;it has to be 3,5,7,11,203; default is 3&#41;

     --set-local-config localConfigurationFile
         set a local configuration file in replacement of conf/jmodeltest.conf

     --set-property propertyName&#61;propertyValue
         set a new value for a property contained in the configuration file &#40;conf/jmodeltest.conf&#41;

     -S NNI|SPR|BEST
         tree topology search operation option &#40;NNI &#40;fast&#41;, SPR &#40;a bit slower&#41;, BEST &#40;best of NNI and SPR&#41;&#41; &#40;default is BEST&#41;

     -t fixed|BIONJ|ML
             base tree for likelihood calculations &#40;e.g., -t BIONJ&#41;
             fixed  &#40;common BIONJ-JC topology&#41;
             BIONJ  &#40;Neighbor-Joining topology&#41;
             ML     &#40;Maximum Likelihood topology&#41; &#40;default&#41;

     -tr numberOfThreads
         number of threads to execute &#40;default is 4&#41;

     -u treeFileName
         user tree for likelihood calculations  &#40;e.g., -u data.tre&#41;

     -uLnL
         calculate delta AIC,AICc,BIC against unconstrained likelihood &#40;e.g., -uLnL&#41;

        &#40;default is false if the input alignment has gaps or ambiguous characters&#41;

     -v
         do model averaging and parameter importances &#40;e.g., -v&#41; &#40;default is false&#41;

     -w
         write PAUP block &#40;e.g., -w&#41; &#40;default is false&#41;

     -z
         strict consensus type for model-averaged phylogeny &#40;e.g., -z&#41; &#40;default is majority rule&#41;

 Command line: java -jar jModeltest.jar -d sequenceFileName &#91;arguments&#93;

 Example: java -jar jModeltest.jar -d sequenceFileName -i -f -g 4 -BIC -AIC -AICc -DT -v -a -w</code></pre>
<h2 id="phymlæ„æ ‘"><a href="#phymlæ„æ ‘" class="header-anchor">phyMLæ„æ ‘</a></h2>
<h3 id="gtrgi_ä¸ç”¨bootstrap_è€Œæ˜¯ç”¨alrt-sh-likeæ£€æµ‹"><a href="#gtrgi_ä¸ç”¨bootstrap_è€Œæ˜¯ç”¨alrt-sh-likeæ£€æµ‹" class="header-anchor">GTR&#43;G&#43;I ä¸ç”¨bootstrap è€Œæ˜¯ç”¨aLRT-SH-likeæ£€æµ‹</a></h3>
<pre><code class="language-sh">phyml -b -4 --run_id GTR&#43;G&#43;I -m GTR -a e -v e -o tlr --no_memory_check -i &lt;**.phy&gt;</code></pre>
<h3 id="gtrg"><a href="#gtrg" class="header-anchor">GTR&#43;G      </a></h3>
<pre><code class="language-sh">phyml -b -4 --run_id GTR&#43;G -m GTR -a e -o tlr --no_memory_check -i &lt;**.phy&gt;</code></pre>
<p>aLRT is a statistical test to compute branch supports. It applies to every &#40;internal&#41; branch and is computed along PhyML run on the original data set. Thus, aLRT is much faster than standard bootstrap which requires running PhyML 100-1,000 times with resampled data sets. As with any test, the aLRT branch support is significant when it is larger than 0.90-0.99. With good quality data &#40;enough signal and sites&#41;, the sets of branches with bootstrap proportion &gt;0.75 and aLRT&gt;0.9 &#40;SH-like option&#41; tend to be similar. Perform bootstrap and number of resampled data sets </p>
<h3 id="æ£€æµ‹å·²æœ‰æ ‘çš„shå€¼"><a href="#æ£€æµ‹å·²æœ‰æ ‘çš„shå€¼" class="header-anchor">æ£€æµ‹å·²æœ‰æ ‘çš„SHå€¼</a></h3>
<p>-u å·²æœ‰æ ‘ -o lr #åªå¯¹æé•¿lå’Œæ›¿ä»£ç‡rè¿›è¡Œä¼˜åŒ–.</p>
<h2 id="raxmlæ„æ ‘"><a href="#raxmlæ„æ ‘" class="header-anchor">RAxMLæ„æ ‘</a></h2>
<h3 id="installation__2"><a href="#installation__2" class="header-anchor">Installation</a></h3>
<p><a href="http://sco.h-its.org/exelixis/software.html">HERE</a>ï¼Œ è¿™ä¸ªç«™ç‚¹è¿˜æœ‰<code>EAxML</code>,å…³æ³¨ä¸€ä¸‹ã€‚</p>
<h3 id="usage__2"><a href="#usage__2" class="header-anchor">USAGE</a></h3>
<h4 id="å¿«é€Ÿå¼€å§‹å«bootstrapæ£€æµ‹"><a href="#å¿«é€Ÿå¼€å§‹å«bootstrapæ£€æµ‹" class="header-anchor">å¿«é€Ÿå¼€å§‹&#40;å«bootstrapæ£€æµ‹&#41;ï¼š</a></h4>
<pre><code class="language-sh">raxml-SSE3 -f a -m GTRGAMMA -p 12345 -# 1000 -x 12345 -n &#91;outname&#93; -s &#91;in_file&#40;phy/fas &#41;&#93;</code></pre>
<h4 id="å¤šçº¿ç¨‹"><a href="#å¤šçº¿ç¨‹" class="header-anchor">å¤šçº¿ç¨‹</a></h4>
<pre><code class="language-sh">raxmlHPC-PTHREADS-SSE3 -T &#91;çº¿ç¨‹æ•°&#93; -f a -m GTRGAMMAI -p 12345 -# 1000 -x 12345 -n &#91;out&#93; -s &#91;IN&#93;</code></pre>
<h2 id="mrbayes_ä½¿ç”¨"><a href="#mrbayes_ä½¿ç”¨" class="header-anchor">mrbayes ä½¿ç”¨</a></h2>
<h3 id="ç®€å•æ­¥éª¤"><a href="#ç®€å•æ­¥éª¤" class="header-anchor">ç®€å•æ­¥éª¤</a></h3>
<p>1  åºåˆ—çš„æ¯”å¯¹ï¼Œç„¶åå°†æ¯”å¯¹å¥½çš„åºåˆ—è½¬åŒ–æˆ.nexæ ¼å¼</p>
<p>2  è¿è¡ŒMrBayesï¼Œç®€å•æ­¥éª¤å¦‚ä¸‹ï¼šï¼ˆä¾æ¬¡è¾“å…¥å‘½ä»¤ï¼Œå®Œæˆç®€å•ä¹Ÿæœ€å¸¸ç”¨çš„åˆ†æï¼‰ï¼š</p>
<pre><code class="language-sh">1 Execute filename.nex #æ‰“å¼€å¾…åˆ†ææ–‡ä»¶ã€‚
2 Lset nst&#61;6 rates&#61;invgamma 
#è¯¥å‘½ä»¤è®¾ç½®è¿›åŒ–æ¨¡å‹ä¸ºwith gamma-distributed rate variation across siteså’Œa proportion of invariable sitesçš„GTRæ¨¡å‹ã€‚æ¨¡å‹å¯æ ¹æ®éœ€è¦æ›´æ”¹ï¼Œä¸è¿‡ä¸€èˆ¬æ— é¡»æ›´æ”¹ã€‚
3  mcmc ngen&#61;1000000 samplefreq&#61;100 printfreq&#61;100 diagnfreq&#61;1000
#This will ensure that you get at least 200 samples from the posterior probability distribution, 
#and that diagnostics are calculated every 1,000 generations. For larger data sets you probably want 
#to run the analysis longer and sample less frequently. The default sample and printfrequency is 500, 
#the default diagnostic frequency is 5,000, and the default run length is 1,000,000. You can find the 
#predicted remaining time to completion of the analysis in the last column printed to screen
 
4  å¦‚æœåˆ†è£‚é¢‘ç‡åˆ†æ”¯é¢‘ç‡split frequenciesçš„æ ‡å‡†åå·®standard deviationåœ¨100,000ä»£generationsä»¥åä½äº0.01ï¼Œ
   å½“ç¨‹åºè¯¢é—®ï¼šâ€œContinue the analysis? &#40;yes/no&#41;â€ï¼Œå›ç­”noï¼›å¦‚æœé«˜äº0.01ï¼Œyesç»§ç»­ç›´åˆ°è¯¥å€¼ä½äº0.01ã€‚
 
5  sump burnin&#61;250 &#40;æˆ–è€…sump relburnin&#61;yes burninfrac&#61;0.25&#41;
#ï¼ˆåœ¨æ­¤ä¸º1000ä¸ªæ ·å“ï¼Œå³ä»»ä½•ç›¸å½“äºä½ å–æ ·çš„25ï¼…çš„å€¼,ä¸åŠ burninå°±ç»Ÿè®¡å…¨éƒ¨ï¼‰
#å‚æ•°æ€»ç»“summarize the parameterï¼Œç¨‹åºä¼šè¾“å‡ºä¸€ä¸ªå…³äºæ ·å“ï¼ˆsampleï¼‰çš„æ›¿ä»£æ¨¡å‹å‚æ•°çš„æ€»ç»“è¡¨ï¼Œ
#åŒ…æ‹¬meanï¼Œmodeå’Œ95 &#37; credibility interval of each parameterï¼Œè¦ä¿è¯æ‰€æœ‰å‚æ•°PSRFï¼ˆthe potential scale
# reduction factorï¼‰çš„å€¼æ¥è¿‘1.0ï¼Œå¦‚æœä¸æ¥è¿‘ï¼Œåˆ†ææ—¶é—´è¦å»¶é•¿ã€‚
 
6  sumt burnin&#61;250ï¼Œ&#40;æˆ–è€…sumt relburnin&#61;yes burninfrac&#61;0.25&#41; 
#æ€»ç»“æ ‘summarize treeã€‚ç¨‹åºä¼šè¾“å‡ºä¸€ä¸ªå…·æœ‰æ¯ä¸€ä¸ªåˆ†æ”¯çš„posterior probabilitiesçš„æ ‘ä»¥åŠä¸€ä¸ªå…·æœ‰å¹³å‡æé•¿
#mean branch lengthsçš„æ ‘ã€‚è¿™äº›æ ‘ä¼šè¢«ä¿å­˜åœ¨ä¸€ä¸ªå¯ä»¥ç”±treeviewç­‰è¯»å–çš„æ ‘æ–‡ä»¶ä¸­</code></pre>
<h2 id="paml-codeml_branch-site_model_test2"><a href="#paml-codeml_branch-site_model_test2" class="header-anchor">paml-codeml: branch-site model &#40;test2&#41;:</a></h2>
<h3 id="å‰æœŸå‡†å¤‡"><a href="#å‰æœŸå‡†å¤‡" class="header-anchor"><ol>
<li><p>å‰æœŸå‡†å¤‡ï¼š</p>
</li>
</ol>
</a></h3>
<p>æ¯”å¯¹å¥½çš„åºåˆ—ï¼ˆæ°¨åŸºé…¸æ¯”å¯¹ï¼‰
æ ‘æ–‡ä»¶ï¼ˆphyæ ¼å¼ï¼Œåªè¦åå­—å’Œæ‹¬å·ï¼Œå‰æ™¯æç”¨â€œ#1â€æ ‡æ³¨ï¼‰
æ§åˆ¶æ–‡ä»¶ï¼ˆå¦‚æœåä¸ºcodemlçš„è¯ è¿è¡Œç¨‹åºæ˜¯æ— éœ€å£°æ˜ï¼Œå…¶ä»–åå­—éœ€è¦å£°æ˜ï¼‰</p>
<h3 id="ol_start2_å‚æ•°é€‰æ‹©"><a href="#ol_start2_å‚æ•°é€‰æ‹©" class="header-anchor"><ol start="2">
<li><p>å‚æ•°é€‰æ‹©ï¼š </p>
</li>
</ol>
</a></h3>
<p>Model Aâ€™, with w2 &#61; 1 fixed
    model &#61; 2 NSsites &#61; 2 fix_omega &#61; 1 omega &#61; 1
Model A, with w2 estimated
    model &#61; 2 NSsites &#61; 2 fix_omega &#61; 0 omega &#61; 1.5 &#40;or any value &gt; 1&#41;</p>


</div>
<!-- article style -->


 <hr>

<script src="https://utteranc.es/client.js"
    repo="songtaogui/blog_comments"
    issue-term="pathname"
    theme="github-light"
    crossorigin="anonymous"
    async>
</script>


<hr> 

<div class="container">
    
    <footer class="site-footer">
        <p class="powered-by">
          <script type="text/javascript" src="//rf.revolvermaps.com/0/0/4.js?i=5jnkqismgmy&amp;m=5&amp;h=128&amp;c=baff00&amp;r=0" async="async"></script><br>
            Total visits: <span id="busuanzi_value_site_pv"></span>.<br>
            Built with <a href="https://franklinjl.org" target="_blank" rel="noopener">Franklin.jl</a> and <a
                href="https://julialang.org" target="_blank" rel="noopener">Julia</a>.<br>
            Texts are licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>.<br>
            Codes are licensed under the <a href="https://mit-license.org/">MIT License</a>.<br>
            <strong>&copy; Songtao Gui</strong>. Last modified: August 04, 2023.
          </p>
    </footer>
    
</div>
</div>


</div>
<!--article container -->
</article>


<!-- CONTENT ENDS HERE -->

 <script src="/libs/highlight/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();hljs.configure({tabReplace: '    '});</script>
 

<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js
    integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js
    integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js
    integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js
    integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script>
<script src="/libs/academic/academic.min.js"></script>
<script src="/libs/pure/ui.js"></script>

</body>


 <a class="js-back-to-top back-to-top-button" href="#">
    <img src="/assets/img/up.svg" style="width: 100px;">
</a>
<!-- <button class="js-back-to-top back-to-top-button" title="BackToTop">ï¸½</button> -->
<script src="https://cdn.staticfile.org/jquery/2.2.4/jquery.min.js"></script>
<script>
    $(function () {
        var $win = $(window);
        var $backToTop = $('.js-back-to-top');
        // å½“ç”¨æˆ·æ»šåŠ¨åˆ°ç¦»é¡¶éƒ¨100åƒç´ æ—¶ï¼Œå±•ç¤ºå›åˆ°é¡¶éƒ¨æŒ‰é’®
        $win.scroll(function () {
            if ($win.scrollTop() > 100) {
                $backToTop.show();
            } else {
                $backToTop.hide();
            }
        });
        // å½“ç”¨æˆ·ç‚¹å‡»æŒ‰é’®æ—¶ï¼Œé€šè¿‡åŠ¨ç”»æ•ˆæœè¿”å›å¤´éƒ¨
        $backToTop.click(function () {
            $('html, body').animate({
                scrollTop: 0
            }, 200);
        });
    });
</script>
<style>
    /* back to top button */
    .back-to-top-button {
        display: none;
        position: fixed;
        width: 50px;
        bottom: 3%;
        right: 3%;
        z-index: 99;
        font-size: 15px;
        border: 0px solid #696969;
        background-color: #fff;
        color: #222222;
        outline: none;
        cursor: pointer;
        padding: 7px;
        border-radius: 40px;
    }

    .back-to-top-button:hover {
        background-color: #c0dbf5;
        color: #fff;
    }
</style> 


<!-- PJAXé…ç½®: å…ˆä¸æäº†! -->
<!--   
  <script src="https://cdn.jsdelivr.net/npm/pjax@0.2.8/pjax.js"></script>
  
  <script>
    var pjax = new Pjax({
      elements: "a",
      selectors: [
        "title",
        "meta[name=description]",
        ".the-header",
        ".the-content",
        ".the-sidebar",
      ]
    })
  </script> -->

</html>



