
<!doctype html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="google-site-verification" content="sKCEBeALSs13uui0cWMviq9hlm0rPByKQKNHnOdq-d0" />
    <meta name="baidu-site-verification" content="code-MBwKQgVpns" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="author" content="Songtao Gui">

    

    
    
    <meta name="keywords" content="MaizeMethylML_note">
    <title>MaizeMethylML_note</title>
    

    <link rel="icon" href="/assets/favicon.svg">

    
<link rel=preconnect href="https://fonts.gstatic.com" crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.1/css/all.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css" crossorigin=anonymous>
<link rel=stylesheet
    href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap"
    crossorigin=anonymous>
<!-- locally served -->
<link rel="stylesheet" href="/css/pure.css">
<link rel=stylesheet href="/libs/academic/academic.min.css">
<link rel="stylesheet" href="/css/layout.css">
<link rel="stylesheet" href="/css/extra.css">
<link rel=stylesheet href="/css/codeset.css">
<link rel=stylesheet href="/css/admonition.css">
<link rel=stylesheet href="/css/cv.css">
<link rel=stylesheet href="/css/search-box.css">
<link rel="stylesheet" href="https://cdn.staticfile.org/font-awesome/4.7.0/css/font-awesome.css">
<!-- <link rel=stylesheet href="/css/font-awesome.css"> -->

<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>

<!-- nutshell + highlight + katex + plotly -->

<link rel=stylesheet href="/libs/highlight/styles/atom-one-light.min.css">






    <!-- Vega -->
    <script src="https://cdn.jsdelivr.net/npm/vega@5.22.1"></script>
    <script src="https://cdn.jsdelivr.net/npm/vega-lite@5.5.0"></script>
    <script src="https://cdn.jsdelivr.net/npm/vega-embed@6.21.0"></script>
    <script src="https://cdn.jsdelivr.net/npm/@svgdotjs/svg.js@3.0/dist/svg.min.js"></script>
    <style media="screen">
        /* Add space between Vega-Embed links  */
        .vega-actions a {
          margin-right: 5px;
        }
    </style>
    <!-- end Vega -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js"
        integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script>
</head>

<body id=top data-spy="scroll" data-offset="70" data-target=#navbar-main>
    <!-- Swiftype -->
    <script type="text/javascript">
        (function(w,d,t,u,n,s,e){w['SwiftypeObject']=n;w[n]=w[n]||function(){
        (w[n].q=w[n].q||[]).push(arguments);};s=d.createElement(t);
        e=d.getElementsByTagName(t)[0];s.async=1;s.src=u;e.parentNode.insertBefore(s,e);
        })(window,document,'script','//s.swiftypecdn.com/install/v2/st.js','_st');
        
        _st('install','J57vdqtTa8oYL-ikpKNG','2.0.0');
    </script>
    
     <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main>
    <div class=container>
        <!-- Main name -->
        <div class="d-none d-lg-inline-flex">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Button for narrow mode -->
        <button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar
            aria-expanded=false aria-label="Toggle navigation">
            <span><i class="fas fa-bars"></i></span>
        </button>
        <!-- Main name for mobile mode -->
        <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Menu items -->
        <div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content>
            <ul class="navbar-nav d-md-inline-flex">
                <li class=nav-item><a class=nav-link href="/posts/"><span>Posts</span></a></li>
                <li class=nav-item><a class=nav-link href="/courses/"><span>Courses</span></a></li>
                <li class=nav-item><a class=nav-link href="/notebooks/"><span>Notebooks</span></a></li>
                <li class=nav-item><a class=nav-link href="/cv/"><span>CurriculumVitae</span></a></li>
                <li class=nav-item><a class=nav-link href="/tag/"><span>Tags</span></a></li>
            </ul>
        </div>
        <!-- <div class="nav-linksmall d-none d-lg-inline-flex inputBox">
                <input type="text" class="st-default-search-input" placeholder="Input then press ENTER" value="" required>
                <div class="search"></div>
            <script>
                var searchForm = document.getElementById('searchForm');
                searchForm.onsubmit = function () {
                    var url = this.action;
                    var q = this.children['q'].value;
                    var q = 'site:songtaogui.github.io+' + q
                    // var site = this.children['site'].value;
                    var url = url + '?q=' + q;
                    window.open(url,'__blank');
                    return false;
                };
            </script>
        </div> -->
        <!-- 基于bing site搜索: -->
        <div class="nav-linksmall d-none d-lg-inline-flex inputBox">
            <form action="https://bing.com/search" id="searchForm">
                <input type="text" name="q" value="" placeholder="Input then press ENTER" value="" required>
                <div class="search"></div>
            </form>
            <script>
                var searchForm = document.getElementById('searchForm');
                searchForm.onsubmit = function () {
                    var url = this.action;
                    var q = this.children['q'].value;
                    var q = 'site:songtaogui.github.io+' + q
                    // var site = this.children['site'].value;
                    var url = url + '?q=' + q;
                    window.open(url,'__blank');
                    return false;
                };
            </script>
        </div>
        <!-- <div class="nav-icon d-none d-lg-inline-flex"> -->
        <div class="nav-icon">
            <a class=nav-linksmall href="/giants" target="_blank"><span><img class="nav-img"
                src="/assets/img/giant.svg"></span></a>
            <a class=nav-linksmall href="/achievements/" target="_blank"><span><img class="nav-img"
                        src="/assets/img/achievement.svg"></span></a>
            <a class=nav-linksmall href="/feed.xml" target="_blank"><span><img class="nav-img"
                        src="/assets/img/rss.svg"></span></a>
            <a class=nav-linksmall href="/musics" target="_blank"><span><img class="nav-img"
                        src="/assets/img/music.svg"></span></a>
        </div>

    </div>
</nav> 

    <span class="js-widget-page d-none"></span>

    
    <article class=article>
        <div class="article-container pt-3">
             <h1>MaizeMethylML_note</h1> 
        </div>
        
        <div class="article-container">
            <!-- 关闭article-style, 不然跟nutshell不兼容 -->
            <!-- <div class="gst-style"> -->
            <div class="layout-main">
                <!-- <div class="article-style"> -->
                
                
                <div class="franklin-content">
<div class="franklin-toc"><ol><li><a href="#甲基化分析流程">甲基化分析流程</a><ol><li><a href="#测序reads比对">测序Reads比对</a></li><li><a href="#鉴定位点的甲基化水平">鉴定位点的甲基化水平</a></li><li><a href="#鉴定高和低甲基化区间">鉴定高和低甲基化区间</a></li><li><a href="#提取序列">提取序列</a></li><li><a href="#给序列添加特征">给序列添加特征</a></li></ol></li><li><a href="#hzs的流程">HZS的流程</a></li></ol></div>
<h1 id="甲基化分析流程"><a href="#甲基化分析流程" class="header-anchor">甲基化分析流程</a></h1>

<fieldset class="hadmon admon-hack"><legend class="hadmon-legend admon-legend-hack"> 流程概览</legend>
</p>
<ul>
<li><p>测序Reads比对</p>
</li>
<li><p>位点甲基化水平鉴定</p>
</li>
<li><p>高/低甲基化区间鉴定</p>
</li>
<li><p>序列提取</p>
</li>
<li><p>添加特征</p>
</li>
</ul>
<p>
</fieldset>

<h2 id="测序reads比对"><a href="#测序reads比对" class="header-anchor">测序Reads比对</a></h2>
<ul>
<li><p>测序Reads原始数据下载, 查阅关联群体甲基化的文章:</p>
<blockquote>
<p>Xu J, Chen G, Hermanson PJ, Xu Q, Sun C, Chen W, et al. Population-level analysis reveals the widespread occurrence and phenotypic consequence of DNA methylation variation not tagged by genetic variation in maize. Genome Biol. 2019;20:243.</p>
</blockquote>
</li>
<li><p>每个材料的甲基化测序reads要比对到<strong>各自</strong>参考基因组, HZS的参考基因组在: <code>/jbyan/stgui/work/09_ZEAMAP_database/00_CNGBDB_V1.0/01_Offical_data/01_genomics/01_genomes/08_Zea_mays_ssp_mays/HZS/Zea_mays_ssp_mays.HZS.v1.0.genome.fa.gz</code></p>
</li>
<li><p>比对工具我用的是<code>bsmap</code>&#40;集群后来数据迁移, 原始环境中的软件删掉了, 请自行安装&#41;, 比对完后的bam需要sort, 具体命令如下:</p>
</li>
</ul>
<div class="indent-block" style="margin-left: 2em !important"> <fieldset class="code code-bash"><legend class="code-legend code-legend-bash">bash</legend></p>
<pre><code class="language-bash">ref&#61;/path/to/your/ref.fa
r1&#61;/path/to/read1.fq.gz
r2&#61;/path/to/read2.fq.gz
pre&#61;&quot;output-prefix&quot;
cpu&#61;4
if &#91; &#33; -s &quot;&#36;&#123;pre&#125;.bsmap.mkdup.bam&quot; &#93;;then
    if &#91; &#33; -s &quot;&#36;&#123;pre&#125;.bsmap.sort.bam&quot; &#93;;then
        if &#91; &#33; -s &quot;&#36;&#123;pre&#125;.bsmap.bam&quot; &#93;;then
            bsmap -a &#36;r1 -b &#36;r2 -d &#36;ref -q 20 -f 5 -p &#36;cpu -v 0.05 -s 16 -S 1 -n 0 |\
            samtools view -@ &#36;cpu -Sb -o &#36;&#123;pre&#125;.bsmap.bam
            if &#91; &#36;? -ne 0 &#93;;then echo &quot;bsmap failed: Non-zero exit&quot; &gt;&amp;2; rm -f &#36;&#123;pre&#125;.bsmap.bam; exit 1;fi
        fi
        sambamba sort -t &#36;cpu -o &#36;&#123;pre&#125;.bsmap.sort.bam &#36;&#123;pre&#125;.bsmap.bam
        if &#91; &#36;? -ne 0 &#93;;then echo &quot;sort bam failed: Non-zero exit&quot; &gt;&amp;2; rm -f &#36;&#123;pre&#125;.bsmap.sort.bam*; exit 1;fi
    fi
    sambamba markdup -t &#36;cpu &#36;&#123;pre&#125;.bsmap.sort.bam &#36;&#123;pre&#125;.bsmap.mkdup.bam &amp;&amp;\
    rm -f &#36;&#123;pre&#125;.bsmap.bam &#36;&#123;pre&#125;.bsmap.sort.bam*
fi</code></pre>
<p><div class="code-lag">bash</div></fieldset></p>
<ul>
<li><p>如果有多个重复的reads, 每个重复的bam还要合并, 用<code>samtools merge</code>即可</p>
</li>
</ul>
<p></div>
<h2 id="鉴定位点的甲基化水平"><a href="#鉴定位点的甲基化水平" class="header-anchor">鉴定位点的甲基化水平</a></h2>
<p>用<code>MethylDackel</code>, 具体命令/参数如下:</p>
<fieldset class="code code-bash"><legend class="code-legend code-legend-bash">bash</legend></p>
<pre><code class="language-bash">ref&#61;/path/to/ref.fa
bam&#61;/path/to/bsmap.sort.bam
cpu&#61;4
MethylDackel extract -q 20 -p 5 -d 3 -@ &#36;cpu --CHG --CHH &#36;ref &#36;bam</code></pre>
<p><div class="code-lag">bash</div></fieldset>
<h2 id="鉴定高和低甲基化区间"><a href="#鉴定高和低甲基化区间" class="header-anchor">鉴定高和低甲基化区间</a></h2>
<p>鉴定高低甲基化区间的标准与2020年NAM<code>Science</code>文章中的标准保持一致:</p>
<blockquote>
<p>To identify UMRs, reference genomes were segmented into 100-bp intervals. Intervals lacking at least four covered CHG-context cytosines &#40;CHGs&#41; were discarded. Coverage was calculated on a per-cytosine basis and summed over each interval, and any interval with less than 20 reads covering CHGs was discarded. Intervals with methylation of greater than 20&#37;, calculated using the weighted methylation formula &#40;Schultz et al. 2012&#41;, were also discarded. This was repeated on 20bp sliding increments, and all overlapping intervals or intervals separated by only 20 bp were merged to define larger UMRs. UMR edges were then trimmed such that their boundaries were defined by CHGs with less than or equal to 20&#37; methylation. At this stage UMRs that overlapped with blacklisted regions &#40;identified based on abnormally high coverage of the 150nt paired end Illumina reads that were used in each genome’s assembly&#41; were discarded. This process was repeated using CG/CHG methylation combined rather than CHG methylation alone and both sets of UMRs were merged. Finally, UMRs that were shorter than 150 bp in length were discarded.</p>
</blockquote>
<p>我写了一个脚本用来流程化以上操作:</p>
<p>源码: <code>/jbyan/stgui/sftw/gst_scripts/gst_hlmr.sh</code></p>
<p><div class="aadmon aadmon-note"> 需要预先安装以下软件才可以使用: <code>bedtools</code>, <code>csvtk</code>, <code>bgzip</code></div> </p>
<p>用法如下&#40;默认参数就是我们甲基化数据的参数&#41;:</p>
<fieldset class="code code-bash"><legend class="code-legend code-legend-bash">bash</legend></p>
<pre><code class="language-bash">------------------------------------------------------------
Get LMR and HMR from input methylation bedgraph
------------------------------------------------------------
Dependence: bedtools, csvtk, bgzip
------------------------------------------------------------
USAGE:
    bash gst_hlmr.sh &#91;OPTIONS&#93;

OPTIONS: &#40;&#91;R&#93;:required  &#91;O&#93;:optional&#41;
# common; I/O
    -h, --help                          show help and exit.
    -i, --in            &lt;str&gt;   &#91;R&#93;     input file in bedgraph format
    -o, --out           &lt;str&gt;   &#91;O&#93;     output prefix name &#40;default: prefix of --in file&#41;
# sliding window
    -w, --window        &lt;int&gt;   &#91;O&#93;     set sliding window size in bp &#40;default: 100&#41;
    -s, --step          &lt;int&gt;   &#91;O&#93;     set step size in bp &#40;default: 20&#41;
# site filtering
    --min_cd            &lt;int&gt;   &#91;O&#93;     the minimum cytosine supported read number to support methylation/un-methylation &#40;default: 3&#41;
    --min_td            &lt;int&gt;   &#91;O&#93;     the minimum cytosine supported read number to support methylation/un-methylation &#40;default: 3&#41;
# window filtering
    --min_rd            &lt;int&gt;   &#91;O&#93;     the minimum number of supported reads in the region &#40;default: 20&#41;
    --min_nc            &lt;int&gt;   &#91;O&#93;     the minimum number of cytosine sites in the region &#40;default: 4&#41;
    --min_mrl           &lt;int&gt;   &#91;O&#93;     the minimum length of methylation region in bp &#40;default: 150&#41;
    --lmr             &lt;1-100&gt;   &#91;O&#93;     set LMR methylation level cut-off from 0 to 100 &#40;&#37;&#41; &#40;default: 20&#41;
    --hmr             &lt;1-100&gt;   &#91;O&#93;     set HMR methylation level cut-off from 0 to 100 &#40;&#37;&#41; &#40;default: 80&#41;
# misc
    --keep                              Kept tmp files
------------------------------------------------------------
Author: Songtao Gui
E-mail: songtaogui@sina.com</code></pre>
<p><div class="code-lag">bash</div></fieldset>
<p>运行该脚本会得到每个合并的区间的甲基化类型&#40;下表中的<code>Type</code>&#41;:</p>
<table><tr><th align="left">#Chr</th><th align="left">Start0</th><th align="left">End</th><th align="left">Type</th><th align="left">Len</th><th align="left">WMF</th><th align="left">Valid<em>CpG</em>num</th><th align="left">Reads<em>mC</em>num</th><th align="left">Reads<em>nC</em>num</th></tr><tr><td align="left">chr1</td><td align="left">140</td><td align="left">42180</td><td align="left">LMR</td><td align="left">42040</td><td align="left">1.1948</td><td align="left">12706</td><td align="left">2093</td><td align="left">173084</td></tr><tr><td align="left">chr1</td><td align="left">42200</td><td align="left">53800</td><td align="left">LMR</td><td align="left">11600</td><td align="left">0.6981</td><td align="left">3222</td><td align="left">360</td><td align="left">51210</td></tr><tr><td align="left">chr1</td><td align="left">53940</td><td align="left">55300</td><td align="left">LMR</td><td align="left">1360</td><td align="left">0.9143</td><td align="left">381</td><td align="left">38</td><td align="left">4118</td></tr><tr><td align="left">chr1</td><td align="left">55840</td><td align="left">56080</td><td align="left">LMR</td><td align="left">240</td><td align="left">0.0000</td><td align="left">21</td><td align="left">0</td><td align="left">84</td></tr><tr><td align="left">chr1</td><td align="left">56460</td><td align="left">57580</td><td align="left">LMR</td><td align="left">1120</td><td align="left">1.0162</td><td align="left">224</td><td align="left">22</td><td align="left">2143</td></tr></table>
<h2 id="提取序列"><a href="#提取序列" class="header-anchor">提取序列</a></h2>
<p>我也写了一个脚本批量根据甲基化区间提取序列, 我们提取序列的标准是:</p>
<ol>
<li><p>按照固定长度滑窗</p>
</li>
<li><p>如果区间小于滑窗长度, 双端补齐N</p>
</li>
</ol>
<p>提序列脚本: <code>/jbyan/stgui/work/05_MethylML/01_Models/00_training_data/00_seqs_and_labels/get_win_seq.sh</code></p>
<p>用法: <fieldset class="code code-bash"><legend class="code-legend code-legend-bash">bash</legend></p>
<pre><code class="language-bash">------------------------------------------------------------
get win seq
------------------------------------------------------------
Dependence: bedtools seqkit
------------------------------------------------------------
USAGE:
    bash get_win_seq.sh &lt;in.bed&#91;.gz&#93;&gt; &lt;ref.fa&#91;.gz&#93;&gt; &lt;win_size&gt; &lt;prefix&gt;
NOTE:
    prefix should be &#91;REF&#93;__&#91;Tissue&#93;__&#91;MethyType&#93;, eg:
    &quot;B73__root__CHG&quot;
------------------------------------------------------------
Author: Songtao Gui
E-mail: songtaogui@sina.com</code></pre>
<p><div class="code-lag">bash</div></fieldset></p>
<p>输入就是上一步鉴定的不同甲基化程度的甲基化区间。</p>
<h2 id="给序列添加特征"><a href="#给序列添加特征" class="header-anchor">给序列添加特征</a></h2>
<p>这个比较复杂,需要提前整理好不同特征的注释文件, 这里先贴上代码路径, 和我生成的文件目录, 需要跑的时候我再细讲解:</p>
<p>代码: <code>/jbyan/stgui/work/05_MethylML/01_Models/00_training_data/01_features/add_features.sh</code></p>
<p>注释文件示例: <code>/jbyan/stgui/work/05_MethylML/01_Models/00_training_data/01_features/&#123;01_gene,02_TEs,03_Exp,04_atac&#125;</code></p>
<h1 id="hzs的流程"><a href="#hzs的流程" class="header-anchor">HZS的流程</a></h1>
<ol>
<li><p>下载HZS的甲基化Reads</p>
</li>
<li><p>比对 -&gt; 排序</p>
</li>
<li><p>鉴定甲基化位点</p>
</li>
<li><p>鉴定甲基化区间</p>
</li>
<li><p>提取序列</p>
</li>
<li><p>添加特征</p>
</li>
<li><p>给诗杰训练</p>
</li>
</ol>


</div>
<!-- article style -->


 <hr>

<script src="https://utteranc.es/client.js"
    repo="songtaogui/blog_comments"
    issue-term="pathname"
    theme="github-light"
    crossorigin="anonymous"
    async>
</script>


<hr> 

<div class="container">
    
    <footer class="site-footer">
        <p class="powered-by">
            Total visits: <span id="busuanzi_value_site_pv"></span>.<br>
            Built with <a href="https://franklinjl.org" target="_blank" rel="noopener">Franklin.jl</a> and <a
                href="https://julialang.org" target="_blank" rel="noopener">Julia</a>.<br>
            The text is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>.<br>
            The code is licensed under the <a href="https://mit-license.org/">MIT License</a>.<br>
            <strong>&copy; Songtao Gui</strong>. Last modified: November 14, 2022.

            <!-- <span class="float-right" aria-hidden="true">
        <a href="#" class="back-to-top">
          <span class="button_icon">
            <i class="fas fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span> -->

        </p>
    </footer>
    
</div>
</div>


</div>
<!--article container -->
</article>


<!-- CONTENT ENDS HERE -->

 <script src="/libs/highlight/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();hljs.configure({tabReplace: '    '});</script>
 

<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js
    integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js
    integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js
    integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js
    integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script>
<script src="/libs/academic/academic.min.js"></script>
<script src="/libs/pure/ui.js"></script>

</body>


 <a class="js-back-to-top back-to-top-button" href="#">
    <img src="/assets/img/up.svg" style="width: 100px;">
</a>
<!-- <button class="js-back-to-top back-to-top-button" title="BackToTop">︽</button> -->
<script src="https://cdn.staticfile.org/jquery/2.2.4/jquery.min.js"></script>
<script>
    $(function () {
        var $win = $(window);
        var $backToTop = $('.js-back-to-top');
        // 当用户滚动到离顶部100像素时，展示回到顶部按钮
        $win.scroll(function () {
            if ($win.scrollTop() > 100) {
                $backToTop.show();
            } else {
                $backToTop.hide();
            }
        });
        // 当用户点击按钮时，通过动画效果返回头部
        $backToTop.click(function () {
            $('html, body').animate({
                scrollTop: 0
            }, 200);
        });
    });
</script>
<style>
    /* back to top button */
    .back-to-top-button {
        display: none;
        position: fixed;
        width: 50px;
        bottom: 3%;
        right: 3%;
        z-index: 99;
        font-size: 15px;
        border: 0px solid #696969;
        background-color: #fff;
        color: #222222;
        outline: none;
        cursor: pointer;
        padding: 7px;
        border-radius: 40px;
    }

    .back-to-top-button:hover {
        background-color: #c0dbf5;
        color: #fff;
    }
</style> 


<!-- PJAX配置: 先不搞了! -->
<!--   
  <script src="https://cdn.jsdelivr.net/npm/pjax@0.2.8/pjax.js"></script>
  
  <script>
    var pjax = new Pjax({
      elements: "a",
      selectors: [
        "title",
        "meta[name=description]",
        ".the-header",
        ".the-content",
        ".the-sidebar",
      ]
    })
  </script> -->

</html>


