<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="author" content="Songtao Gui">
  <meta name="description" content="Head of Research">
  <meta name="theme-color" content="#2962ff">
  <meta name="google-site-verification" content="sKCEBeALSs13uui0cWMviq9hlm0rPByKQKNHnOdq-d0" />
  
  <link rel="icon" href="/assets/favicon.png">
  <link rel="stylesheet" href="https://cdn.staticfile.org/font-awesome/4.7.0/css/font-awesome.css">
  <link
  rel=preconnect
  href="https://fonts.gstatic.com"
  crossorigin=anonymous>

<link
  rel=stylesheet
  href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.1/css/all.min.css"  crossorigin=anonymous>

<link
  rel=stylesheet
  href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css"
  crossorigin=anonymous>

<link
  rel=stylesheet
  href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css"
  crossorigin=anonymous>

<link
  rel=stylesheet
  href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap"
  crossorigin=anonymous>

<!-- locally served -->
<link rel="stylesheet" href="/css/pure.css">
<link rel=stylesheet href="/libs/academic/academic.min.css">
<link rel="stylesheet" href="/css/layout.css">
<link rel="stylesheet" href="/css/extra.css">
<link rel=stylesheet href="/css/codeset.css">
<link rel=stylesheet href="/css/admonition.css">
<link rel=stylesheet href="/css/cv.css">

<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>

<!-- nutshell + highlight + katex + plotly -->

<link rel=stylesheet href="/libs/highlight/styles/atom-one-light.min.css">





  <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js" 
  integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script>
   <title>Irys-scaffolding-pipeline</title>  


</head>


<body id=top data-spy="scroll" data-offset="70" data-target=#navbar-main>

   <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main>
  <div class=container>
    <!-- Main name -->
    <div class="d-none d-lg-inline-flex">
      <a class=navbar-brand href="/">Songtao Gui</a>
    </div>
    <!-- Button for narrow mode -->
    <button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar aria-expanded=false aria-label="Toggle navigation">
      <span><i class="fas fa-bars"></i></span>
    </button>
    <!-- Main name for mobile mode -->
    <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
      <a class=navbar-brand href="/">Songtao Gui</a>
    </div>
    <!-- Menu items -->
    <div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content>
      <ul class="navbar-nav d-md-inline-flex">
        <li class=nav-item><a class=nav-link href="/posts/"><span>Posts</span></a></li>
        <!-- <li class=nav-item><a class=nav-link href="/notebooks/"><span>Notebooks</span></a></li> -->
        <li class=nav-item><a class=nav-link href="/courses/"><span>Courses</span></a></li>
        <li class=nav-item><a class=nav-link href="/achievements/"><span>Achievements</span></a></li>
        <li class=nav-item><a class=nav-link href="/cv/"><span>CurriculumVitae</span></a></li>
        <li class=nav-item><a class=nav-link href="/tag/"><span>Tags</span></a></li>
        <!-- <li class=nav-item><a class=nav-link href="/giants/"><span>Giants</span></a></li> -->


        <!-- 
        <li class=nav-item><a class=nav-link ><span> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></a></li>
        <li class=nav-item>
          <a class=nav-link>
            <span>
              <script src="/libs/lunr/lunr.js"></script>
              <script src="lunr.stemmer.support.js"></script>
              <script src="/libs/lunr/lunr.zh.js"></script>
              <script src="/libs/lunr/lunr_index.js"></script>
              <script src="/libs/lunr/lunrclient.min.js"></script>
              <form id="lunrSearchForm" name="lunrSearchForm">
                <input class="search-input" name="q" placeholder="Enter search term" type="text">
                <input type="submit" value="Search" formaction="/search/index.html">
              </form>
            </span>
          </a>
        </li>
        -->

      </ul>
    </div>
    <div class="nav-icon d-none d-lg-inline-flex">
      <a class=nav-linksmall href="/notebooks" target="_blank"><span><img class="nav-img" src="/assets/img/notebook.svg"></span></a>
      <a class=nav-linksmall href="/giants" target="_blank"><span><img class="nav-img" src="/assets/img/giant.svg"></span></a>
      <a class=nav-linksmall href="/feed.xml" target="_blank"><span><img class="nav-img" src="/assets/img/rss.svg"></span></a>
    </div>
    <div class="nav-icon ">
      <a class=nav-linksmall href="/musics" target="_blank"><span><img class="nav-img" src="/assets/img/music.svg"></span></a>
    </div>
    
  </div>
</nav>
 

  <span class="js-widget-page d-none"></span>
  
  
    <article class=article>
      <div class="article-container pt-3">
        <h1>Irys-scaffolding-pipeline</h1>
      </div>
      
      <div class="article-container">
        <!-- 关闭article-style, 不然跟nutshell不兼容 -->
        <!-- <div class="gst-style"> -->
        <div class="layout-main">
        <!-- <div class="article-style"> -->
  
<div class="franklin-content"><p> <div class="tags"><a href="/tag/" id="tag-icon"><svg width="20" height="20" viewBox="0 0 512 512"><defs><style>.cls-1{fill:#141f38}</style></defs><path class="cls-1" d="M215.8 512a76.1 76.1 0 0 1-54.17-22.44L22.44 350.37a76.59 76.59 0 0 1 0-108.32L242 22.44A76.11 76.11 0 0 1 296.2 0h139.2A76.69 76.69 0 0 1 512 76.6v139.19A76.08 76.08 0 0 1 489.56 270L270 489.56A76.09 76.09 0 0 1 215.8 512zm80.4-486.4a50.69 50.69 0 0 0-36.06 14.94l-219.6 219.6a51 51 0 0 0 0 72.13l139.19 139.19a51 51 0 0 0 72.13 0l219.6-219.61a50.67 50.67 0 0 0 14.94-36.06V76.6a51.06 51.06 0 0 0-51-51zm126.44 102.08A38.32 38.32 0 1 1 461 89.36a38.37 38.37 0 0 1-38.36 38.32zm0-51a12.72 12.72 0 1 0 12.72 12.72 12.73 12.73 0 0 0-12.72-12.76z"/><path class="cls-1" d="M217.56 422.4a44.61 44.61 0 0 1-31.76-13.16l-83-83a45 45 0 0 1 0-63.52L211.49 154a44.91 44.91 0 0 1 63.51 0l83 83a45 45 0 0 1 0 63.52L249.31 409.24a44.59 44.59 0 0 1-31.75 13.16zm-96.7-141.61a19.34 19.34 0 0 0 0 27.32l83 83a19.77 19.77 0 0 0 27.31 0l108.77-108.7a19.34 19.34 0 0 0 0-27.32l-83-83a19.77 19.77 0 0 0-27.31 0l-108.77 108.7z"/><path class="cls-1" d="M294.4 281.6a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85zM256 320a12.75 12.75 0 0 1-9.05-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2A12.8 12.8 0 0 1 256 320zM217.6 358.4a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 1 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85z"/></svg></a><a href="/tag/bioinformatics/">bioinformatics</a>, <a href="/tag/bionano/">bionano</a>&nbsp;&nbsp;&nbsp;&nbsp;<a id="tag-icon"><svg width="20" height="20" class="icon" viewBox="0 0 1024 1024" version="1.1" p-id="3825" width="64" height="64"><path d="M512 416a96 96 0 1 0 0 192 96 96 0 0 0 0-192z m511.952 102.064c-0.016-0.448-0.064-0.864-0.096-1.296a8.16 8.16 0 0 0-0.08-0.656c0-0.32-0.064-0.624-0.128-0.928-0.032-0.368-0.064-0.736-0.128-1.088-0.032-0.048-0.032-0.096-0.032-0.144a39.488 39.488 0 0 0-10.704-21.536c-32.672-39.616-71.536-74.88-111.04-107.072-85.088-69.392-182.432-127.424-289.856-150.8-62.112-13.504-124.576-14.064-187.008-2.64-56.784 10.384-111.504 32-162.72 58.784-80.176 41.92-153.392 99.696-217.184 164.48-11.808 11.984-23.552 24.224-34.288 37.248-14.288 17.328-14.288 37.872 0 55.216 32.672 39.616 71.52 74.848 111.04 107.056 85.12 69.392 182.448 127.408 289.888 150.784 62.096 13.504 124.608 14.096 187.008 2.656 56.768-10.4 111.488-32 162.736-58.768 80.176-41.936 153.376-99.696 217.184-164.48 11.792-12 23.536-24.224 34.288-37.248 5.712-5.872 9.456-13.44 10.704-21.568l0.032-0.128a12.592 12.592 0 0 0 0.128-1.088c0.064-0.304 0.096-0.624 0.128-0.928l0.08-0.656 0.096-1.28c0.032-0.656 0.048-1.296 0.048-1.952l-0.096-1.968zM512 704c-106.032 0-192-85.952-192-192s85.952-192 192-192 192 85.968 192 192c0 106.048-85.968 192-192 192z" p-id="3826"></path></svg></a><span id="busuanzi_value_page_pv"></span></div></p>
<h2 id="github_上的_bionano组装流程"><a href="#github_上的_bionano组装流程" class="header-anchor">GitHub 上的 Bionano组装流程</a></h2>
<p><a href="https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/blob/master/KSU_bioinfo_lab/assemble_XeonPhi/assemble_XeonPhi_LAB.md">GitHub</a></p>
<h2 id="程序的安装和配置"><a href="#程序的安装和配置" class="header-anchor"><ol>
<li><p>程序的安装和配置</p>
</li>
</ol>
</a></h2>
<h3 id="11_install_bionano_tools_and_scripts"><a href="#11_install_bionano_tools_and_scripts" class="header-anchor">1.1 Install BioNano tools and scripts:</a></h3>
<p>参考Bionano IrysView® v2.3官方安装说明：<a href="http://www.bnxinstall.com/training/docs/IrysView_Software_Installation_Guide_23.pdf">HERE</a></p>
<p>另， Bionano IrysView® v2.3 Software Training Guide 在： <a href="http://www.bnxinstall.com/training/docs/IrysView_Software_Training_Guide_23.pdf">HERE</a></p>
<pre><code class="language-shell">wget http://www.bnxinstall.com/RefalignerAssembler/Linux/AVX/tools_v4287_AVX.tar.gz
wget http://www.bnxinstall.com/scripts/v4125/scripts_v4125.tar.gz
#然后都解压到 &#36;SFTW/app/BioNano 目录下，都设定成可执行</code></pre>
<h3 id="12_git_install_the_ksu_custom_software_using_the_code_below"><a href="#12_git_install_the_ksu_custom_software_using_the_code_below" class="header-anchor">1.2 Git install the KSU custom software using the code below:</a></h3>
<pre><code class="language-shell">cd ~
git clone https://github.com/i5K-KINBRE-script-share/Irys-scaffolding.git
git clone https://github.com/i5K-KINBRE-script-share/BNGCompare.git

#将以下目录中的*.pl添加执行权限后，添加到PATH：
&quot;XXXXXX/Irys-scaffolding/KSU_bioinfo_lab/map_tools/&quot;
&quot;XXXXXX/Irys-scaffolding/KSU_bioinfo_lab/assemble_XeonPhi/&quot;
&quot;XXXXXX/Irys-scaffolding/KSU_bioinfo_lab/assemble_XeonPhi/third-party/&quot;</code></pre>
<p>为了方便使用，添加环境变量到<code>.bashrc</code>：</p>
<blockquote>
<p>export KSU&#61;/XXX/XXX/Irys-scaffolding/KSU<em>bioinfo</em>lab</p>
</blockquote>
<h3 id="13_personal_edit"><a href="#13_personal_edit" class="header-anchor">1.3 Personal Edit</a></h3>
<p><strong>注意！！！</strong></p>
<blockquote>
<p>该pipeline是编写用于XeonPhi服务器和CL集群的，会用到SGE等集群运算的东西，默认参数在普通服务器上跑，</p>
</blockquote>
<p>会占用全部cpu并且报错。 而且pipeline识别的默认Bionano程序是安装在~目录下，如果你的程序不在~目录, 也需要改。</p>
<h4 id="131_change_for_nomal_linux"><a href="#131_change_for_nomal_linux" class="header-anchor">1.3.1 change for nomal linux</a></h4>
<h5><code>rescale_stretch.pl</code></h5>
<p>修改<code>&#36;KSU/assemble_XeonPhi/rescale_stretch.pl</code> 文件，主要修改:</p>
<ol>
<li><p>tools和scripts目录的路径: </p>
</li>
<li><p>piplineCL.py 的运行参数：</p>
<ul>
<li><p>Threads 改小</p>
</li>
<li><p>-C 参数去掉</p>
</li>
</ul>
</li>
</ol>
<p>修改的<code>rescale_stretch.pl</code>:</p>
<pre><code class="language-julia">#&#33;/usr/bin/perl
################################################################################
#   
#	USAGE: perl rescale_stretch.pl &lt;bnx_dir&gt; &lt;p_value Threshold&gt; &lt;project&gt; &lt;reference&gt;
#
#  Created by jennifer shelton
#  perl rescale_stretch.pl &#36;bnx_dir &#36;reference &#36;T &#36;dirname
#
################################################################################
use strict;
use warnings;
use File::Basename; # enable maipulating of the full path
# use List::Util qw&#40;max&#41;;
# use List::Util qw&#40;sum&#41;;
################################################################################
##############           Customize RefAligner Settings        ##################
################################################################################
# This pipeline was designed to run on a Xeon Phi server with 576 cores &#40;48x12-core Intel Xeon CPUs&#41;, 256GB of RAM, and Linux CentOS 7 operating system. Customization of this section may be required to run the BioNano Assembler on a different machine. Customization of Irys-scaffolding/KSU_bioinfo_lab/assemble_XeonPhi/clusterArguments.xml may also be required for assembly to run successfully on a different cluster.
my &#36;refaligner &#61; &#36;ENV&#123;&quot;HOME&quot;&#125; .&quot;/GuiSongtao/software/BioNano/tools/RefAligner&quot;; # See &quot;~/tools/RefAligner -help&quot; for details about these parameters
unless &#40;-f &#36;refaligner&#41;
&#123;
    die &quot;Can&#39;t find RefAligner at &#36;refaligner. Please add correct path to rescale_stretch.pl and retry:\n &#36;&#33;&quot;;
&#125;
my &#36;maxmem &#61; 40; #gst edited, raw is 240
my &#36;hashmaxmem &#61; 70; # gst edited, raw is 120
my &#36;insertThreads &#61; 10; # gst edited, raw is 16
my &#36;maxthreads &#61; 15; # gst edited, raw is 64
################################################################################
##############                 get arguments                  ##################
################################################################################
my &#36;assembly_directory&#61;&#36;ARGV&#91;0&#93;;
my &#36;T&#61;&#36;ARGV&#91;1&#93;;
my &#36;project &#61; &#36;ARGV&#91;2&#93;;
my &#36;de_novo&#61;1; # project is de novo
if &#40;&#36;ARGV&#91;3&#93;&#41;
&#123;
    if &#40;-f &quot;&#36;ARGV&#91;3&#93;&quot;&#41;
    &#123;
        &#36;de_novo&#61;0; # change project is not de novo because a refernce CMAP exists
        print &quot;de_novo &#61; false\n&quot;;
    &#125;
&#125;
else
&#123;
    print &quot;de_novo &#61; true\n&quot;;
&#125;
my &#36;reference;
if&#40;&#36;de_novo &#61;&#61; 0&#41;
&#123;
    &#36;reference &#61; &#36;ARGV&#91;3&#93;;
    print &quot;ref &#61; &#36;ARGV&#91;3&#93;\n&quot;;
&#125;
my &#36;dirname &#61; dirname&#40;__FILE__&#41;;
################################################################################
############         Adjust stretch &#40;bpp&#41; for BNX files       ##################
################################################################################
my &#36;bnx_list_file &#61; &quot;&#36;&#123;assembly_directory&#125;/bnx_list.text&quot;; #creat list to use in merging BNX files
open &#40;my &#36;bnx_list, &quot;&gt;&quot;, &#36;bnx_list_file&#41; or die &quot;Can&#39;t open &#36;bnx_list_file: &#36;&#33;&quot;;
my &#36;bnx_dir &#61; &quot;&#36;assembly_directory/bnx&quot;;
opendir&#40;DIR, &#36;bnx_dir&#41; or die &quot;Can&#39;t open &#36;bnx_dir&#33;\n&quot;; # open directory full of .bnx files
while &#40;my &#36;file &#61; readdir&#40;DIR&#41;&#41;
&#123;
    next if &#40;&#36;file &#61;~ m/^\./&#41;; # ignore files beginning with a period
    next if &#40;&#36;file &#33;~ m/Molecules.*\.bnx&#36;/&#41;; # if the filename is Molecules.bnx
    ############################################################################
    ############              Check BNX file version          ##################
    ############################################################################
    my &#36;bnx_stats&#61;&#96;perl &#36;&#123;dirname&#125;/../map_tools/bnx_version.pl -i &#36;&#123;bnx_dir&#125;/&#36;&#123;file&#125;&#96;;
    print &quot;&#36;bnx_stats&quot;;
    &#36;bnx_stats &#61;~ /version: &#40;.*&#41;\n/;
    if &#40;&#36;1 &lt; 1&#41;
    &#123;
        die &quot;Exiting because this workflow was developed for BNX version 1.0&#43;&quot;;
    &#125;
    print &#36;bnx_list &quot;&#36;&#123;bnx_dir&#125;/&#36;&#123;file&#125;\n&quot;;
&#125;
################################################################################
############                Merge your BNX files               #################
################################################################################
my &#36;refalign_log_file &#61; &quot;&#36;&#123;assembly_directory&#125;/refAlign_log.txt&quot;; #creat log for refAligner output
open &#40;my &#36;refalign_log, &quot;&gt;&quot;, &#36;refalign_log_file&#41; or die &quot;Can&#39;t open &#36;refalign_log_file: &#36;&#33;&quot;;

mkdir &quot;&#36;&#123;assembly_directory&#125;/all_flowcells&quot;; # Make an outout directory for merged flowcells

my &#36;merge_bnxs &#61; &#96;&#36;refaligner -if &#36;bnx_list_file -o &#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged -merge -bnx -minsites 5 -minlen 100 -maxthreads &#36;maxthreads&#96;;
print &#36;refalign_log &quot;&#36;merge_bnxs&quot;;
my &#36;merged_file &#61; &quot;&#36;&#123;bnx_dir&#125;/../all_flowcells/bnx_merged.bnx&quot;;
if&#40;&#36;de_novo &#61;&#61; 0&#41;
&#123;
    ############################################################################
    ####         Subsample 50,000 molecules and run alignment with          ####
    ####         very loose alignment parameters &#40;T should be about         ####
    ####                   inverse of the genome size&#41;.                     ####
    ############################################################################
    ## Subsample 50,000 molecules and run alignment with very loose alignment parameters.
    my &#36;error_A &#61; &quot;&#36;&#123;bnx_dir&#125;/../all_flowcells/bnx_merged_errA&quot;;
    my &#36;get_error_A &#61; &#96;&#36;refaligner -o &#36;error_A -i &#36;merged_file -ref &#36;reference -minlen 180 -minsites 9 -refine 0 -id 1 -mres 0.9 -res 3.4 -resSD 0.75 -FP 1.0 -FN 0.1 -sf 0.2 -sd 0 -sr 0.02 -resbias 4.0 64 -outlier 1e-4 -endoutlier 1e-4 -S -999 -T 1e-4 -MapRate 0.7 -A 5 -nosplit 2 -biaswt 0 -deltaX 4 -deltaY 6 -extend 1 -PVres 2 -f -randomize -subset 1 50000 -BestRef 1 -BestRefPV 1 -hashoffset 1 -AlignRes 1.5 -resEstimate -M 5 -hashgen 5 3 2.4 1.5 0.05 5.0 1 1 2 -hash -hashdelta 10 -maxmem &#36;maxmem -hashmaxmem &#36;hashmaxmem -insertThreads &#36;insertThreads -maxthreads &#36;maxthreads&#96;; ## TEST when you get a reference &#33;&#33;&#33;&#33;&#33;&#33;&#33;&#33;
    print &#36;refalign_log &quot;#####\n#####\nStep 1: Subsample 50,000 molecules and run alignment with very loose alignment parameters\n#####\n#####\n&quot;;
    print &#36;refalign_log &quot;&#36;get_error_A&quot;;

    ## The error metrics returned are refined in the following step using 100000 molecules and more stringent alignments.
    my &#36;error_B &#61; &quot;&#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged_errB&quot;;

    my &#36;get_error_B &#61; &#96;&#36;refaligner -o &#36;error_B -i &#36;merged_file -ref &#36;reference -readparameters &#36;&#123;error_A&#125;_id1.errbin -minlen 180 -minsites 9 -refine 0 -id 1 -resbias 4.0 64 -outlier 1e-4 -endoutlier 1e-4 -S -999 -T 1e-4 -MapRate 0.7 -A 5 -nosplit 2 -biaswt 0 -deltaX 4 -deltaY 6 -extend 1 -PVres 2 -f -randomize -subset 1 100000 -BestRef 1 -BestRefPV 1 -hashoffset 1  -AlignRes 1.5 -resEstimate -M 5 -hashgen 5 3 2.4 1.5 0.05 5.0 1 1 2 -hash -hashdelta 10 -maxmem &#36;maxmem -hashmaxmem &#36;hashmaxmem -insertThreads &#36;insertThreads  -maxthreads &#36;maxthreads&#96;;
    print &#36;refalign_log &quot;#####\n#####\nStep 2: The error metrics returned are refined in the following step using 100000 molecules and more stringent alignments.\n#####\n#####\n&quot;;

    print &#36;refalign_log &quot;&#36;get_error_B&quot;;

    ## Finally the original BNX set is rescaled per the noise parameters from the second step. In this step, after noise parameters have be estimated using long molecules the minimum molecule length is set back to 100 kb.

    my &#36;merged_file_adjusted &#61; &quot;&#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged_adj&quot;;

    #my &#36;get_adjusted_bnx &#61; &#96;~/tools/RefAligner -o &#36;merged_file_adjusted -i &#36;merged_file -ref &#36;reference -readparameters &#36;&#123;error_B&#125;_id1.errbin -minlen 100 -minsites 9 -refine 0 -id 1 -resbias 4.0 64 -outlier 1e-4 -endoutlier 1e-4 -S -9 -T 1e-4 -MapRate 0.7 -A 5 -nosplit 2 -biaswt 0 -deltaX 4 -deltaY 6 -extend 1 -PVres 2 -f -BestRef 1 -BestRefPV 1 -maptype 1 -hashoffset 1 -AligneRes 1.5  -resEstimate -ScanScaling 2 -M 5 -hashgen 5 3 2.4 1.5 0.05 5.0 1 1 2 -hash -hashdelta 10 -maxmem 240 -hashmaxmem 120 -insertThreads 16  -maxthreads 64&#96;; ## threw error unknown option:-AligneRes&#40;argc&#61;28&#41;

    my &#36;get_adjusted_bnx &#61; &#96;&#36;refaligner -o &#36;merged_file_adjusted -i &#36;merged_file -ref &#36;reference -readparameters &#36;&#123;error_B&#125;_id1.errbin -minlen 100 -minsites 9 -refine 0 -id 1 -resbias 4.0 64 -outlier 1e-4 -endoutlier 1e-4 -S -9 -T 1e-4 -MapRate 0.7 -A 5 -nosplit 2 -biaswt 0 -deltaX 4 -deltaY 6 -extend 1 -PVres 2 -f -BestRef 1 -BestRefPV 1 -maptype 1 -hashoffset 1 -resEstimate -ScanScaling 2 -M 5 -hashgen 5 3 2.4 1.5 0.05 5.0 1 1 2 -hash -hashdelta 10 -maxmem &#36;maxmem -hashmaxmem &#36;hashmaxmem -insertThreads &#36;insertThreads  -maxthreads &#36;maxthreads&#96;;
    print &#36;refalign_log &quot;#####\n#####\nStep 3: Finally the original BNX set is rescaled per the noise parameters from the second step. In this step, after noise parameters have be estimated using long molecules the minimum molecule length is set back to 100 kb\n#####\n#####\n&quot;;
    print &#36;refalign_log &quot;&#36;get_adjusted_bnx&quot;;

    ## Run final alignment for noise parameters .
    my &#36;rescaled_file &#61; &quot;&#36;&#123;bnx_dir&#125;/../all_flowcells/bnx_merged_adj_rescaled.bnx&quot;;
    my &#36;error_C &#61; &quot;&#36;&#123;bnx_dir&#125;/../all_flowcells/bnx_merged_rescaled_final&quot;;
    my &#36;get_error_C &#61; &#96;&#36;refaligner -o &#36;error_C -i &#36;rescaled_file -ref &#36;reference -nosplit 2 -M 5 -biaswt 0 -Mfast 0 -FP 1.5 -FN 0.15 -sf 0.2 -sd 0.2 -A 5 -res 3.5 -resSD 0.7 -outlier 1e-4 -endoutlier 1e-4 -minlen 100 -minsites 5 -T &#36;T -insertThreads &#36;insertThreads -maxthreads &#36;maxthreads -randomize 1 -subset 1 10000&#96;;
    print &#36;refalign_log &quot;#####\n#####\nStep 4: Subsample 10,000 molecules and run final alignment for noise parameters\n#####\n#####\n&quot;;
    print &#36;refalign_log &quot;&#36;get_error_C&quot;;

    close&#40;&#36;refalign_log&#41;;

    ############################################################################
    ####                      Plot rescaling factors by scan                ####
    ############################################################################
    #  scan&#61;0:RunIndex&#61;1,ScanNumber&#61;0:scale&#61;0.99938102
    open &#40;&#36;refalign_log, &quot;&lt;&quot;, &#36;refalign_log_file&#41; or die &quot;Can&#39;t open &#36;refalign_log_file: &#36;&#33;&quot;;

    my &#36;rescaling_factor_list_out_file &#61; &quot;&#36;&#123;assembly_directory&#125;/&#36;&#123;project&#125;/bnx_rescaling_factors.tab&quot;;
    open &#40;my &#36;rescaling_factor_list_out, &quot;&gt;&quot;, &#36;rescaling_factor_list_out_file&#41; or die &quot;Can&#39;t open &#36;rescaling_factor_list_out_file in rescale_stretch.pl\n&quot;;
    print &#36;rescaling_factor_list_out &quot;flow_cell\tscan\tscale\n&quot;;
    my &#36;scan_count&#61;0;
    my &#36;first_scans &#61; &#39;&#39;;
    while &#40;&lt;&#36;refalign_log&gt;&#41; # grab rescaling factor for each scan from RefAlign log
    &#123;
        chomp;
        if &#40;/scan&#61;&#40;.*&#41;:RunIndex&#61;&#40;.*&#41;,ScanNumber&#61;&#40;.*&#41;:scale&#61;&#40;.*&#41;/&#41;
        &#123;
            my &#36;scan &#61; &#36;1;
            my &#36;RunIndex &#61; &#36;2;
            my &#36;ScanNumber &#61; &#36;3;
            my &#36;scale &#61; &#36;4;
            print &#36;rescaling_factor_list_out &quot;&#36;RunIndex\t&#36;ScanNumber\t&#36;scale\n&quot;;
            &#43;&#43;&#36;scan_count;
            if &#40;&#36;ScanNumber &#61;&#61; 0&#41;
            &#123;
                &#36;first_scans &#61; &#36;first_scans.&quot; &quot;.&#36;scan_count; # grab position of the first scan for each BNX file
            &#125;
        &#125;
    &#125;

    my &#36;command &#61; &quot;Rscript &#36;&#123;dirname&#125;/plot_bnx_rescaling_factors.R &#36;&#123;assembly_directory&#125;/&#36;&#123;project&#125;/bnx_rescaling_factors.tab &#36;&#123;assembly_directory&#125;/&#36;&#123;project&#125;/bnx_rescaling_factors.pdf &#36;&#123;scan_count&#125;&#36;&#123;first_scans&#125;&quot;;
    #print &quot;Command: &#36;command\n&quot;;
    my &#36;get_rescaled_bnx &#61; &#96;&#36;command&#96;; # plot rescaling factor for each scan
    print &quot;&#36;get_rescaled_bnx&quot;;

    ############################################################################

    print &quot;Done merging BNX files and rescaling molecule maps\n&quot;;
&#125;
else
&#123;
    print &quot;Done merging BNX files\n&quot;;
&#125;</code></pre>
<h5 id="assemblepl"><a href="#assemblepl" class="header-anchor"><code>assemble.pl</code></a></h5>
<p>修改 <code>&#36;KSUb/assemble_XeonPhi/assemble.pl</code> 文件：</p>
<ol>
<li><p>RefAligner路径</p>
</li>
<li><p>RefAligner 运行参数</p>
</li>
</ol>
<pre><code class="language-julia">#&#33;/usr/bin/perl
################################################################################
#   
#	USAGE: perl assemble.pl &lt;assembly_directory&gt; &lt;p_value Threshold&gt; &lt;project prefix&gt; &lt;genome size&gt; &lt;reference&gt;
#
#  Created by Jennifer Shelton
#
################################################################################
use strict;
use warnings;
# use List::Util qw&#40;max&#41;;
# use List::Util qw&#40;sum&#41;;
#use XML::Simple;
#use Data::Dumper;
use File::Basename; # enable manipulating of the full path
################################################################################
##############       Customize BioNano Script Settings        ##################
################################################################################
# This pipeline was designed to run on a Xeon Phi server with 576 cores &#40;48x12-core Intel Xeon CPUs&#41;, 256GB of RAM, and Linux CentOS 7 operating system. Customization of this section may be required to run the BioNano Assembler on a different machine. Customization of Irys-scaffolding/KSU_bioinfo_lab/assemble_XeonPhi/clusterArguments.xml may also be required for assembly to run successfully on a different cluster.
my &#36;pipelineCL &#61; &#36;ENV&#123;&quot;HOME&quot;&#125; .&quot;/GuiSongtao/software/BioNano/scripts/pipelineCL.py&quot;; #Change if not ~/scripts/pipelineCL.py #GST changed&#33; raw is:
#my &#36;pipelineCL &#61; &#36;ENV&#123;&quot;HOME&quot;&#125; .&quot;/scripts/pipelineCL.py&quot;; 
unless &#40;-f &#36;pipelineCL&#41;
&#123;
    die &quot;Can&#39;t find pipelineCL.py at &#36;pipelineCL . Please add correct path to assemble.pl and retry:\n &#36;&#33;&quot;;
&#125;
my &#36;tools &#61; &#36;ENV&#123;&quot;HOME&quot;&#125; .&quot;/GuiSongtao/software/BioNano/tools/&quot;; #Change if not ~/tools #GST changed&#33; raw is:
#my &#36;tools &#61; &#36;ENV&#123;&quot;HOME&quot;&#125; .&quot;/tools/&quot;;
unless &#40;-d &#36;tools&#41;
&#123;
    die &quot;Can&#39;t find the BioNano directory \&quot;tools\&quot; at &#36;tools . Please add correct path to assemble.pl and retry:\n &#36;&#33;&quot;;
&#125;
################################################################################
##############                     get arguments              ##################
################################################################################
my &#36;assembly_directory &#61; &#36;ARGV&#91;0&#93;;
my &#36;T &#61; &#36;ARGV&#91;1&#93;;
my &#36;project &#61; &#36;ARGV&#91;2&#93;;
my &#36;genome &#61; &#36;ARGV&#91;3&#93;;
my &#36;de_novo&#61;1; # project is de novo
if &#40;&#36;ARGV&#91;4&#93;&#41;
&#123;
    if &#40;-f &quot;&#36;ARGV&#91;4&#93;&quot;&#41;
    &#123;
        &#36;de_novo&#61;0; # change project is not de novo because a reference CMAP exists
        print &quot;de_novo &#61; false\n&quot;;

    &#125;
&#125;
else
&#123;
    print &quot;de_novo &#61; true\n&quot;;
&#125;
my &#36;ref;
unless&#40;&#36;de_novo &#61;&#61; 1&#41;
&#123;
    &#36;ref &#61; &#36;ARGV&#91;4&#93;;
    print &quot;ref &#61; &#36;ARGV&#91;4&#93;\n&quot;;
&#125;
print &quot;assembly_directory &#61; &#36;ARGV&#91;0&#93;\n&quot;;
print &quot;T &#61; &#36;ARGV&#91;1&#93;\n&quot;;
my &#36;dirname &#61; dirname&#40;__FILE__&#41;; # has no trailing slash
################################################################################
##############              get parameters for XML            ##################
################################################################################
my &#36;T_relaxed &#61; &#36;T * 10;
my &#36;T_strict &#61; &#36;T/10;
my &#40;&#36;FP,&#36;FN,&#36;SiteSD_Kb,&#36;ScalingSD_Kb_square,&#36;LabelDensity&#41;;
my &#36;merged_error_file &#61; &quot;&#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged_rescaled_final.err&quot;;
#0Iteration	1FP&#40;/100kb&#41;	2FNrate	3SiteSD&#40;Kb&#41;	4ScalingSD&#40;Kb^1/2&#41;	5bpp	6res&#40;pixels&#41;	7Maps	8Log10LR&#40;/Maps&#41;	9GoodMaps	10log10LR&#40;/GoodMaps&#41;	11bppSD	12FPrate	13RelativeSD	14ResolutionSD	15LabelDensity&#40;/100kb&#41;	16resSD	17mres	18mresSD
unless&#40;&#36;de_novo &#61;&#61; 1&#41;
&#123;
    open &#40;my &#36;merged_error, &quot;&lt;&quot;, &#36;merged_error_file&#41; or die &quot;Can&#39;t open &#36;merged_error_file&#33;\n&quot;;
    while &#40;&lt;&#36;merged_error&gt;&#41;
    &#123;
        if &#40;/^4\t/&#41;
        &#123;
            &#40;&#36;LabelDensity&#41;&#61;&#40;split/\t/&#41;&#91;15&#93;; # because the final label density is reported as 0
        &#125;
        elsif &#40;/^5\t/&#41;
        &#123;
            &#40;&#36;FP,&#36;FN,&#36;SiteSD_Kb,&#36;ScalingSD_Kb_square&#41;&#61;&#40;split/\t/&#41;&#91;1,2,3,4&#93;;
        &#125;
    &#125;
    print &quot;Label Density for aligned molecule maps per 100 kb: &#36;LabelDensity\n&quot;;
&#125;
################################################################################
##############  Select optArguments.xml file based on genome size ##############
################################################################################
my &#36;xml_infile;
my &#36;iterations &#61; 5;
if &#40; &#36;genome &lt; 100 &#41;
&#123;
    &#36;xml_infile &#61; &quot;&#36;&#123;dirname&#125;/optArguments_small.xml&quot;;
&#125;
elsif &#40; &#36;genome &lt; 1000 &#41;
&#123;
    &#36;xml_infile &#61; &quot;&#36;&#123;dirname&#125;/optArguments_medium.xml&quot;;
&#125;
else
&#123;
    &#36;xml_infile &#61; &quot;&#36;&#123;dirname&#125;/optArguments_human.xml&quot;;
    &#36;iterations &#61; 2; # Can lower the number of itereations for large genomes in future if needed

&#125;
################################################################################
##############                 parse XML                      ##################
################################################################################
my &#37;p_value &#61; &#40;
    &#39;default_t_150&#39; &#61;&gt; &quot;&#36;T&quot;,
    &#39;relaxed_t_150&#39; &#61;&gt; &quot;&#36;T_relaxed&quot;,
    &#39;strict_t_150&#39; &#61;&gt; &quot;&#36;T_strict&quot;,
    &#39;default_t_100&#39; &#61;&gt; &quot;&#36;T&quot;,
    &#39;relaxed_t_100&#39; &#61;&gt; &quot;&#36;T_relaxed&quot;,
    &#39;strict_t_100&#39; &#61;&gt; &quot;&#36;T_strict&quot;,
    &#39;default_t_180&#39; &#61;&gt; &quot;&#36;T&quot;,
    &#39;relaxed_t_180&#39; &#61;&gt; &quot;&#36;T_relaxed&quot;,
    &#39;strict_t_180&#39; &#61;&gt; &quot;&#36;T_strict&quot;
&#41;;
my @commands &#61; qw/default_t_150 relaxed_t_150 strict_t_150 default_t_100 relaxed_t_100 strict_t_100 default_t_180 relaxed_t_180 strict_t_180/;
unless&#40;&#36;de_novo &#61;&#61; 1&#41; # if the project is not de novo then make one assembly to test the effect of the default noise parameters
&#123;
    &#36;p_value&#123;&#39;default_t_default_noise&#39;&#125; &#61; &quot;&#36;T&quot;;
    push&#40;@commands, &#39;default_t_default_noise&#39;&#41;;
&#125;
open &#40;my &#36;out_assemble, &#39;&gt;&#39;,&quot;&#36;&#123;assembly_directory&#125;/assembly_commands.sh&quot;&#41;; # for assembly commands
##################################################################
##############        Write bash scripts        ##################
##################################################################
print &#36;out_assemble &quot;#&#33;/bin/bash\n&quot;;
print &#36;out_assemble &quot;##################################################################\n&quot;;
print &#36;out_assemble &quot;#####                  ASSEMBLY COMMANDS                     #####\n&quot;;
print &#36;out_assemble &quot;##################################################################\n&quot;;
for my &#36;stringency &#40;@commands&#41;
&#123;
    my &#36;current_p_value &#61; &#36;p_value&#123;&#36;stringency&#125;;
    ##################################################################
    ##############     Create assembly directories  ##################
    ##################################################################
    my &#36;out_dir &#61; &quot;&#36;&#123;assembly_directory&#125;/&#36;&#123;stringency&#125;&quot;;
    unless&#40;mkdir &#36;out_dir&#41;
    &#123;
		die &quot;Exiting because unable to create &#36;out_dir\n&quot;;
	&#125;
    ##################################################################
    ##############        Set assembly parameters   ##################
    ##################################################################
    my &#36;xml_final &#61; &quot;&#36;&#123;assembly_directory&#125;/&#36;&#123;stringency&#125;/&#36;&#123;stringency&#125;_final_optArguments.xml&quot;;
    my &#36;min_length &#61; 150;
    unless &#40;&#36;stringency eq &#39;default_t_default_noise&#39;&#41;# skip adjusting molecule length parameter based on the assembly name for the default_noise assembly
    &#123;
        &#36;stringency &#61;~ /.*_t_&#40;.*&#41;/;
        &#36;min_length &#61; &#36;1;
    &#125;
    open &#40;my &#36;optarg_final, &#39;&gt;&#39;, &#36;xml_final&#41; or die &quot;Can&#39;t open &#36;xml_final\n&quot;;
    open &#40;my &#36;optarg, &#39;&lt;&#39;, &#36;xml_infile &#41; or die &quot;Can&#39;t open &#36;xml_infile\n&quot;;
    
    CUSTOMXML: while &#40;&lt;&#36;optarg&gt;&#41;
    &#123;
        if &#40;/&lt;flag attr&#61;\&quot;-T\&quot;.*group&#61;\&quot;Initial Assembly\&quot;/&#41;
        &#123;
            s/&#40;val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display.*group&#61;\&quot;Initial Assembly\&quot;.*&#41;/&#36;1&#36;p_value&#123;&#36;stringency&#125;&#36;3/;
            print &#36;optarg_final &quot;&#36;_&quot;;
#            print &quot;Yes#6\n&quot;;
            next CUSTOMXML;
        &#125;
        elsif &#40;/&lt;flag attr&#61;\&quot;-T\&quot;.*group&#61;\&quot;Extension and Refinement\&quot;/&#41;
        &#123;
            my &#36;new_p&#61;&#36;p_value&#123;&#36;stringency&#125;/10;
            s/&#40;val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display.*group&#61;\&quot;Extension and Refinement\&quot;.*&#41;/&#36;1&#36;&#123;new_p&#125;&#36;3/;
            print &#36;optarg_final &quot;&#36;_&quot;;
#            print &quot;Yes#7\n&quot;;
            next CUSTOMXML;
        &#125;
        elsif &#40;/&lt;flag attr&#61;\&quot;-T\&quot;.*group&#61;\&quot;Merge\&quot;/&#41;
        &#123;
            my &#36;final_p&#61;&#36;p_value&#123;&#36;stringency&#125;/10000;
            s/&#40;val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display.*group&#61;\&quot;Merge\&quot;.*&#41;/&#36;1&#36;&#123;final_p&#125;&#36;3/;
            print &#36;optarg_final &quot;&#36;_&quot;;
#            print &quot;Yes#8\n&quot;;
            next CUSTOMXML;
        &#125;
        elsif &#40;/&lt;flag attr&#61;\&quot;-minlen\&quot;.*group&#61;\&quot;BNX Sort\&quot;/&#41;
        &#123;
#            &lt;flag attr&#61;&quot;-minlen&quot;      val0&#61;&quot;150&quot; display&#61;&quot;Molecule Length Threshold &#40;Kb&#41;&quot; group&#61;&quot;BNX Sort&quot; default0&#61;&quot;150&quot; description&#61;&quot;Minimum length of molecules &#40;kb&#41; that are used in BNX sort. This will also be the minimum length used for all downstream Pipeline stages &#40;entire assembly&#41;.&quot; /&gt;
            s/&#40;.*val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display&#61;\&quot;Molecule Length Threshold.*&#41;/&#36;1&#36;&#123;min_length&#125;&#36;3/;
            print &#36;optarg_final &quot;&#36;_&quot;;
#            print &quot;Yes#1\n&quot;;
            next CUSTOMXML;
        &#125;
        elsif &#40;&#36;stringency ne &#39;default_t_default_noise&#39;&#41;# skip adjusting molecule length and noise parameters for the default_noise assembly or any de novo assembly
        &#123;
            if &#40;&#36;de_novo &#61;&#61; 0&#41;
            &#123;

                if &#40;/&lt;flag attr&#61;\&quot;-FP\&quot;.*group&#61;\&quot;DeNovo Assembly Noise\&quot;/&#41;
                &#123;
                    s/&#40;&lt;flag attr&#61;\&quot;-FP\&quot; val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display.*&#41;/&#36;1&#36;&#123;FP&#125;&#36;3/;
                    print &#36;optarg_final &quot;&#36;_&quot;;
#                    print &quot;Yes#2\n&quot;;
                    next CUSTOMXML;
                &#125;
                elsif &#40;/&lt;flag attr&#61;\&quot;-FN\&quot;.*group&#61;\&quot;DeNovo Assembly Noise\&quot;/&#41;
                &#123;
                    s/&#40;&lt;flag attr&#61;\&quot;-FN\&quot; val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display.*&#41;/&#36;1&#36;&#123;FN&#125;&#36;3/;
                    print &#36;optarg_final &quot;&#36;_&quot;;
#                    print &quot;Yes#3\n&quot;;
                    next CUSTOMXML;
                &#125;
                elsif &#40;/&lt;flag attr&#61;\&quot;-sd\&quot;.*group&#61;\&quot;DeNovo Assembly Noise\&quot;/&#41;
                &#123;
                    s/&#40;val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display.*&#41;/&#36;1&#36;&#123;ScalingSD_Kb_square&#125;&#36;3/;
                    print &#36;optarg_final &quot;&#36;_&quot;;
#                    print &quot;Yes#4\n&quot;;
                    next CUSTOMXML;
                &#125;
                elsif &#40;/&lt;flag attr&#61;\&quot;-sf\&quot;.*group&#61;\&quot;DeNovo Assembly Noise\&quot;/&#41;
                &#123;
                    s/&#40;val0&#61;\&quot;&#41;&#40;.*&#41;&#40;\&quot;\s&#43;display.*&#41;/&#36;1&#36;&#123;SiteSD_Kb&#125;&#36;3/;
                    print &#36;optarg_final &quot;&#36;_&quot;;
#                    print &quot;Yes#5\n&quot;;
                    next CUSTOMXML;
                &#125;
            &#125;
        &#125;
        print &#36;optarg_final &quot;&#36;_&quot;; # print any unchanged line
    &#125;
    ##################################################################
    ##############        Write assembly command    ##################
    ##################################################################
    print &#36;out_assemble &quot;##################################################################\n&quot;;
    print &#36;out_assemble &quot;#####             ASSEMBLY: &#36;&#123;stringency&#125;                \n&quot;;
    print &#36;out_assemble &quot;##################################################################\n&quot;;
#    /home/mic_common/scripts_fresh/pipelineCL.py -y -d -U -T 240 -N 6 -j 240 -i 5 -l /home/irys/data/human_100x_test/output/ -t /usr/src/genome_grid/configurations/Human_hg19/tools -C /home/irys/data/human_100x_test/clusterArguments.xml -b /home/irys/data/human_100x_test/input.bnx -r /home/irys/data/human_100x_test/ref.cmap -a /home/irys/data/human_100x_test/optArguments.xml
    if &#40;&#36;min_length &#33;&#61; 150&#41;
    &#123;
        print &#36;out_assemble &quot;#&quot;; #start with default min length assemblies by commenting other assemblies
    &#125;
    if &#40;&#36;de_novo &#61;&#61; 1&#41;
    &#123;
    		print &#36;out_assemble &quot;python2 &#36;pipelineCL -T 15 -j 15 -N 6 -i &#36;iterations -a &#36;xml_final -w -t &#36;tools -l &#36;out_dir -b &#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged.bnx -V 1 -e &#36;&#123;project&#125;_&#36;&#123;stringency&#125; -p 0 -U\n&quot;; # GST edited&#33; removed &quot; -C &#36;&#123;dirname&#125;/clusterArguments.xml&quot;, and change threads Raw is below:
        #print &#36;out_assemble &quot;python2 &#36;pipelineCL -T 240 -j 240 -N 6 -i &#36;iterations -a &#36;xml_final -w -t &#36;tools -l &#36;out_dir -b &#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged.bnx -V 1 -e &#36;&#123;project&#125;_&#36;&#123;stringency&#125; -p 0 -U -C &#36;&#123;dirname&#125;/clusterArguments.xml\n&quot;;
    &#125;
    else
    &#123;		
    		print &#36;out_assemble &quot;python2 &#36;pipelineCL -T 15 -j 15 -N 6 -i &#36;iterations -a &#36;xml_final -w -t &#36;tools -l &#36;out_dir -b &#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged_adj_rescaled.bnx -V 1 -e &#36;&#123;project&#125;_&#36;&#123;stringency&#125; -p 0 -r &#36;ref -U\n&quot;;  # GST edited&#33; removed &quot; -C &#36;&#123;dirname&#125;/clusterArguments.xml&quot;, and change threads. Raw is below:
        #print &#36;out_assemble &quot;python2 &#36;pipelineCL -T 240 -j 240 -N 6 -i &#36;iterations -a &#36;xml_final -w -t &#36;tools -l &#36;out_dir -b &#36;&#123;assembly_directory&#125;/all_flowcells/bnx_merged_adj_rescaled.bnx -V 1 -e &#36;&#123;project&#125;_&#36;&#123;stringency&#125; -p 0 -r &#36;ref -U -C &#36;&#123;dirname&#125;/clusterArguments.xml\n&quot;;
    &#125;
    if &#40;&#36;min_length &#33;&#61; 150&#41;
    &#123;
        print &#36;out_assemble &quot;#&quot;; #start with default min length assemblies by commenting other assemblies
    &#125;
    print &#36;out_assemble &quot;bash &#36;&#123;dirname&#125;/rm_intermediate_files.sh &#36;&#123;assembly_directory&#125; &#36;&#123;stringency&#125; &#36;&#123;project&#125;\n&quot;;
&#125;

print &quot;Done writing assembly scripts.\n&quot;;</code></pre>
<h4 id="132_add_your_own_enzyme"><a href="#132_add_your_own_enzyme" class="header-anchor">1.3.2 Add your own Enzyme</a></h4>
<h5 id="nick_densitypl"><a href="#nick_densitypl" class="header-anchor"><code>nick_density.pl</code></a></h5>
<p><code>vim &#36;KSU/map_tools/nick_density.pl</code> :</p>
<pre><code class="language-julia">#line 57
my @enzymes &#61; qw/BspQI BbvCI BsmI BsrDI bseCI BssSI/; #BssSI是新加的</code></pre>
<h5 id="fa2cmap_multipl"><a href="#fa2cmap_multipl" class="header-anchor"><code>fa2cmap_multi.pl</code></a></h5>
<p><code>vim &#36;KSU/assemble_XeonPhi/third-party/fa2cmap_multi.pl</code>:</p>
<pre><code class="language-julia"># start from line 47 
my &#37;enzyme &#61; &#40;
	&quot;BspQI&quot; &#61;&gt; &quot;GCTCTTC&quot;,
	&quot;BbvCI&quot; &#61;&gt; &quot;CCTCAGC&quot;,
	&quot;BsmI&quot;  &#61;&gt; &quot;GAATGC&quot;,
	&quot;BsrDI&quot; &#61;&gt; &quot;GCAATG&quot;,
	&quot;bseCI&quot; &#61;&gt; &quot;ATCGAT&quot;,
	&quot;BssSI&quot; &#61;&gt; &quot;CACGAG&quot; #added by GST, 名字要与nick_density.pl中的更改一致
	# You can add more enzymes here ...
	
&#41;;</code></pre>
<h5 id="sewing_machinepl"><a href="#sewing_machinepl" class="header-anchor"><code>sewing_machine.pl</code></a></h5>
<p><code>vim &#36;KSU/stitch/sewing_machine.pl</code>:</p>
<pre><code class="language-julia">#line 163
unless&#40;&#36;enzyme &#61;~ /&#40;BspQI|BbvCI|BsrDI|bseCI|BsmI|BssSI&#41;/&#41; #gst added |BssSI 与前两个更改一致</code></pre>
<h2 id="ol_start2_流程操作"><a href="#ol_start2_流程操作" class="header-anchor"><ol start="2">
<li><p>流程操作</p>
</li>
</ol>
</a></h2>
<h3 id="21_hybrid_scaffold_流程"><a href="#21_hybrid_scaffold_流程" class="header-anchor">2.1 Hybrid Scaffold 流程</a></h3>
<p>具体流程解释见<a href="https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/blob/master/KSU_bioinfo_lab/stitch/sewing_machine_LAB.md">GitHub</a></p>
<p>需要的文件:</p>
<ul>
<li><p>参考基因组Scaffold的fasta序列: <code>ref.fasta</code></p>
</li>
<li><p><code>ref.fasta</code>用对应酶模拟酶切的cmap文件: <code>ref.cmap</code></p>
</li>
<li><p>Bionano组装结果cmap文件:<code>REFINEFINAL1.cmap</code></p>
</li>
</ul>
<p><strong>STEP 1. Compare your assembled genome maps to your reference in silico maps with sewing machine</strong> </p>
<pre><code class="language-julia">perl &#36;KSU/stitch/sewing_machine.pl -help #查看具体参数</code></pre>
<p>基本用法:</p>
<pre><code class="language-julia">perl &#36;KSU/stitch/sewing_machine.pl -o &#91;输出文件夹&#93; -g &#91;Bionano组装结果cmap文件.cmap&#93; -p &#91;项目名称&#93; -e &#91;酶&#93; -f &#91;ref.fasta&#93; -r &#91;ref.cmp&#93;</code></pre>
<p><strong>STEP 2. Choose your best alignment parameters and explore your results</strong></p>
<p>第一步中的&#91;输出文件夹&#93;下<code>.csv</code>包含组装参数,选择你认为最好的参数,用<code>write_report.pl</code>生成最终结果: &#40;以 默认参数为例&#41;</p>
<pre><code class="language-julia">perl &#36;KSU/assemble_XeonPhi/write_report.pl -o &#91;输出文件夹&#93; -g &#91;Bionano组装结果cmap文件.cmap&#93; -p &#91;项目名称&#93; -e &#91;酶&#93; -f &#91;ref.fasta&#93; -r &#91;ref.cmp&#93; --alignment_parameters default_alignment</code></pre>
<p><strong>STEP 3. Explore your results in IrysView</strong></p>
<p>在&#91;输出文件夹&#93;下有</p>
<ul>
<li><p>report.txt 文件</p>
</li>
<li><p>&#91;项目名称&#93; 文件夹</p>
</li>
<li><p>项目名称.tar.gz 压缩包</p>
</li>
</ul>
<p>可根据<a href="https://github.com/i5K-KINBRE-script-share/Irys-scaffolding/blob/master/KSU_bioinfo_lab/assemble_XeonPhi/README.pdf">README.PDF</a> 在windows上用IrysView查看结果</p>

</div> 
<!-- article style -->


 <hr>

<script src="https://utteranc.es/client.js"
    repo="songtaogui/blog_comments"
    issue-term="pathname"
    theme="github-light"
    crossorigin="anonymous"
    async>
</script>


<hr> 

<div class="container">
  <footer class="site-footer">
    <p class="powered-by">
      Total visits: <span id="busuanzi_value_site_pv"></span>.<br>
      Built with <a href="https://franklinjl.org" target="_blank" rel="noopener">Franklin.jl</a> and <a href="https://julialang.org" target="_blank" rel="noopener">Julia</a>.<br>
      The text is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>.<br>
      The code is licensed under the <a href="https://mit-license.org/">MIT License</a>.<br>
      <strong>&copy; Songtao Gui</strong>. Last modified: September 13, 2022. 

      <!-- <span class="float-right" aria-hidden="true">
        <a href="#" class="back-to-top">
          <span class="button_icon">
            <i class="fas fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span> -->

    </p>
  </footer>
</div>
</div>
  </div> <!--article container -->
</article>


<!-- CONTENT ENDS HERE -->
    
     <script src="/libs/highlight/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();hljs.configure({tabReplace: '    '});</script>
 
    <script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin=anonymous></script>
    
  <script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script>

  <script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script>
  
  <script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script>
  
  <script src="/libs/academic/academic.min.js"></script>
  <script src="/libs/pure/ui.js"></script>
  
  </body>


   <a class="js-back-to-top back-to-top-button" href="#">
  <img src="/assets/img/up.svg" style="width: 100px;">
</a>
<!-- <button class="js-back-to-top back-to-top-button" title="BackToTop">︽</button> -->
<script src="https://cdn.staticfile.org/jquery/2.2.4/jquery.min.js"></script>
<script>
  $(function() {
    var $win = $(window);
    var $backToTop = $('.js-back-to-top');
    // 当用户滚动到离顶部100像素时，展示回到顶部按钮
    $win.scroll(function() {
      if ($win.scrollTop() > 100) {
        $backToTop.show();
      } else {
        $backToTop.hide();
      }
    });
    // 当用户点击按钮时，通过动画效果返回头部
    $backToTop.click(function() {
      $('html, body').animate({
        scrollTop: 0
      }, 200);
    });
  });
</script>
<style>
  /* back to top button */
  .back-to-top-button {
      display: none;
      position: fixed;
      width: 50px;
      bottom: 3%;
      right: 3%;
      z-index: 99;
      font-size: 15px;
      border: 0px solid #696969;
      background-color: #fff;
      color: #222222;
      outline: none;
      cursor: pointer;
      padding: 7px;
      border-radius: 40px;
  }
  .back-to-top-button:hover {
      background-color: #c0dbf5;
      color: #fff;
  }
</style>
 
  

  <!-- PJAX配置: 先不搞了! -->
<!--   
  <script src="https://cdn.jsdelivr.net/npm/pjax@0.2.8/pjax.js"></script>
  
  <script>
    var pjax = new Pjax({
      elements: "a",
      selectors: [
        "title",
        "meta[name=description]",
        ".the-header",
        ".the-content",
        ".the-sidebar",
      ]
    })
  </script> -->

</html>
