
<!doctype html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="google-site-verification" content="sKCEBeALSs13uui0cWMviq9hlm0rPByKQKNHnOdq-d0" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="author" content="Songtao Gui">

    
    <meta name="description" content="Brief usage of TRF: a tandem repeats finder
">
    

    
    
    <meta name="keywords" content="trf-usage">
    <title>trf-usage</title>
    

    <link rel="icon" href="/assets/favicon.svg">
    <link rel="stylesheet" href="https://cdn.staticfile.org/font-awesome/4.7.0/css/font-awesome.css">
    <link rel=preconnect href="https://fonts.gstatic.com" crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.1/css/all.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css" crossorigin=anonymous>
<link rel=stylesheet
    href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap"
    crossorigin=anonymous>

<!-- locally served -->
<link rel="stylesheet" href="/css/pure.css">
<link rel=stylesheet href="/libs/academic/academic.min.css">
<link rel="stylesheet" href="/css/layout.css">
<link rel="stylesheet" href="/css/extra.css">
<link rel=stylesheet href="/css/codeset.css">
<link rel=stylesheet href="/css/admonition.css">
<link rel=stylesheet href="/css/cv.css">

<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>

<!-- nutshell + highlight + katex + plotly -->

<link rel=stylesheet href="/libs/highlight/styles/atom-one-light.min.css">



    <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js"
        integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script>
</head>

<body id=top data-spy="scroll" data-offset="70" data-target=#navbar-main>

     <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main>
    <div class=container>
        <!-- Main name -->
        <div class="d-none d-lg-inline-flex">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Button for narrow mode -->
        <button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar
            aria-expanded=false aria-label="Toggle navigation">
            <span><i class="fas fa-bars"></i></span>
        </button>
        <!-- Main name for mobile mode -->
        <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Menu items -->
        <div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content>
            <ul class="navbar-nav d-md-inline-flex">
                <li class=nav-item><a class=nav-link href="/posts/"><span>Posts</span></a></li>
                <!-- <li class=nav-item><a class=nav-link href="/notebooks/"><span>Notebooks</span></a></li> -->
                <li class=nav-item><a class=nav-link href="/courses/"><span>Courses</span></a></li>
                <li class=nav-item><a class=nav-link href="/achievements/"><span>Achievements</span></a></li>
                <li class=nav-item><a class=nav-link href="/cv/"><span>CurriculumVitae</span></a></li>
                <li class=nav-item><a class=nav-link href="/tag/"><span>Tags</span></a></li>
                <!-- <li class=nav-item><a class=nav-link href="/giants/"><span>Giants</span></a></li> -->


                <!-- 
        <li class=nav-item><a class=nav-link ><span> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></a></li>
        <li class=nav-item>
          <a class=nav-link>
            <span>
              <script src="/libs/lunr/lunr.js"></script>
              <script src="lunr.stemmer.support.js"></script>
              <script src="/libs/lunr/lunr.zh.js"></script>
              <script src="/libs/lunr/lunr_index.js"></script>
              <script src="/libs/lunr/lunrclient.min.js"></script>
              <form id="lunrSearchForm" name="lunrSearchForm">
                <input class="search-input" name="q" placeholder="Enter search term" type="text">
                <input type="submit" value="Search" formaction="/search/index.html">
              </form>
            </span>
          </a>
        </li>
        -->

            </ul>
        </div>
        <div class="nav-icon d-none d-lg-inline-flex">
            <a class=nav-linksmall href="/notebooks" target="_blank"><span><img class="nav-img"
                        src="/assets/img/notebook.svg"></span></a>
            <a class=nav-linksmall href="/giants" target="_blank"><span><img class="nav-img"
                        src="/assets/img/giant.svg"></span></a>
            <a class=nav-linksmall href="/feed.xml" target="_blank"><span><img class="nav-img"
                        src="/assets/img/rss.svg"></span></a>
        </div>
        <div class="nav-icon ">
            <a class=nav-linksmall href="/musics" target="_blank"><span><img class="nav-img"
                        src="/assets/img/music.svg"></span></a>
        </div>

    </div>
</nav> 

    <span class="js-widget-page d-none"></span>

    
    <article class=article>
        <div class="article-container pt-3">
             <h1>trf-usage</h1> 
        </div>
        
        <div class="article-container">
            <!-- 关闭article-style, 不然跟nutshell不兼容 -->
            <!-- <div class="gst-style"> -->
            <div class="layout-main">
                <!-- <div class="article-style"> -->
                
                
                <div class="franklin-content"><p> <div class="tags"><a href="/tag/" id="tag-icon"><svg width="20" height="20" viewBox="0 0 512 512"><defs><style>.cls-1{fill:#141f38}</style></defs><path class="cls-1" d="M215.8 512a76.1 76.1 0 0 1-54.17-22.44L22.44 350.37a76.59 76.59 0 0 1 0-108.32L242 22.44A76.11 76.11 0 0 1 296.2 0h139.2A76.69 76.69 0 0 1 512 76.6v139.19A76.08 76.08 0 0 1 489.56 270L270 489.56A76.09 76.09 0 0 1 215.8 512zm80.4-486.4a50.69 50.69 0 0 0-36.06 14.94l-219.6 219.6a51 51 0 0 0 0 72.13l139.19 139.19a51 51 0 0 0 72.13 0l219.6-219.61a50.67 50.67 0 0 0 14.94-36.06V76.6a51.06 51.06 0 0 0-51-51zm126.44 102.08A38.32 38.32 0 1 1 461 89.36a38.37 38.37 0 0 1-38.36 38.32zm0-51a12.72 12.72 0 1 0 12.72 12.72 12.73 12.73 0 0 0-12.72-12.76z"/><path class="cls-1" d="M217.56 422.4a44.61 44.61 0 0 1-31.76-13.16l-83-83a45 45 0 0 1 0-63.52L211.49 154a44.91 44.91 0 0 1 63.51 0l83 83a45 45 0 0 1 0 63.52L249.31 409.24a44.59 44.59 0 0 1-31.75 13.16zm-96.7-141.61a19.34 19.34 0 0 0 0 27.32l83 83a19.77 19.77 0 0 0 27.31 0l108.77-108.7a19.34 19.34 0 0 0 0-27.32l-83-83a19.77 19.77 0 0 0-27.31 0l-108.77 108.7z"/><path class="cls-1" d="M294.4 281.6a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85zM256 320a12.75 12.75 0 0 1-9.05-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2A12.8 12.8 0 0 1 256 320zM217.6 358.4a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 1 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85z"/></svg></a><a href="/tag/bioinformatics/">bioinformatics</a>, <a href="/tag/genomics/">genomics</a>&nbsp;&nbsp;&nbsp;&nbsp;<a id="tag-icon"><svg width="20" height="20" class="icon" viewBox="0 0 1024 1024" version="1.1" p-id="3825" width="64" height="64"><path d="M512 416a96 96 0 1 0 0 192 96 96 0 0 0 0-192z m511.952 102.064c-0.016-0.448-0.064-0.864-0.096-1.296a8.16 8.16 0 0 0-0.08-0.656c0-0.32-0.064-0.624-0.128-0.928-0.032-0.368-0.064-0.736-0.128-1.088-0.032-0.048-0.032-0.096-0.032-0.144a39.488 39.488 0 0 0-10.704-21.536c-32.672-39.616-71.536-74.88-111.04-107.072-85.088-69.392-182.432-127.424-289.856-150.8-62.112-13.504-124.576-14.064-187.008-2.64-56.784 10.384-111.504 32-162.72 58.784-80.176 41.92-153.392 99.696-217.184 164.48-11.808 11.984-23.552 24.224-34.288 37.248-14.288 17.328-14.288 37.872 0 55.216 32.672 39.616 71.52 74.848 111.04 107.056 85.12 69.392 182.448 127.408 289.888 150.784 62.096 13.504 124.608 14.096 187.008 2.656 56.768-10.4 111.488-32 162.736-58.768 80.176-41.936 153.376-99.696 217.184-164.48 11.792-12 23.536-24.224 34.288-37.248 5.712-5.872 9.456-13.44 10.704-21.568l0.032-0.128a12.592 12.592 0 0 0 0.128-1.088c0.064-0.304 0.096-0.624 0.128-0.928l0.08-0.656 0.096-1.28c0.032-0.656 0.048-1.296 0.048-1.952l-0.096-1.968zM512 704c-106.032 0-192-85.952-192-192s85.952-192 192-192 192 85.968 192 192c0 106.048-85.968 192-192 192z" p-id="3826"></path></svg></a><span id="busuanzi_value_page_pv"></span></div></p>
<h2 id="trf安装"><a href="#trf安装" class="header-anchor">TRF安装</a></h2>
<p>wget http://tandem.bu.edu/trf/downloads/trf409.linux64
chmod a&#43;x trf409.linux64
mv trf409.linux64 &#36;SFTW/bin/trf</p>
<h2 id="trf使用"><a href="#trf使用" class="header-anchor">TRF使用</a></h2>
<p>Using Command Line Version of Tandem Repeats Finder</p>
<p>Once the program is installed you can run it with no parameters to obtain information on proper  usage syntax.</p>
<p>If you installed the program as trf then by typing trf at the command line you will see the  following output: </p>
<pre><code class="language-julia">Please use: trf File Match Mismatch Delta PM PI Minscore MaxPeriod &#91;options&#93;

Where: &#40;all weights, penalties, and scores are positive&#41;
  File &#61; sequences input file
  Match  &#61; matching weight
  Mismatch  &#61; mismatching penalty
  Delta &#61; indel penalty
  PM &#61; match probability &#40;whole number&#41;
  PI &#61; indel probability &#40;whole number&#41;
  Minscore &#61; minimum alignment score to report
  MaxPeriod &#61; maximum period size to report
  &#91;options&#93; &#61; one or more of the following:
        -m        masked sequence file
        -f        flanking sequence
        -d        data file
        -h        suppress html output
        -r        no redundancy elimination
        -l &lt;n&gt;    maximum TR length expected &#40;in millions&#41; &#40;eg, -l 3 or -l&#61;3 for 3 million&#41;

Note the sequence file should be in FASTA format:

&gt;Name of sequence
aggaaacctgccatggcctcctggtgagctgtcctcatccactgctcgctgcctctccag
atactctgacccatggatcccctgggtgcagccaagccacaatggccatggcgccgctgt
actcccacccgccccaccctcctgatcctgctatggacatggcctttccacatccctgtg</code></pre>
<p>The program accepts a minimum of eight parameters. Options can be specified to generate additional files.</p>
<p>The following is a more detailed description of the parameters:</p>
<blockquote>
<p><strong>File:</strong> The sequence file to be analyzed in FASTA format&#40; see for details&#41;. Multiple sequence in  the same file are allowed.</p>
<p><strong>Match, Mismatch, and Delta:</strong> Weights for match, mismatch and indels. These parameters are for  Smith-Waterman style local alignment using wraparound dynamic programming. Lower weights allow  alignments with more mismatches and indels. A match weight of 2 has proven effective with mismatch  and indel penalties in the range of 3 to 7. Mismatch and indel weights are interpreted as negative   numbers. A 3 is more permissive and a 7 less permissive. The recomended values for Match Mismatch    and Delta are 2, 7, and 7 respectively.</p>
<p><strong>PM and PI:</strong> Probabilistic data is available for PM values of 80 and 75 and PI values of 10 and </p>
<ol start="20">
<li><p>The best performance can be achieved with values of PM&#61;80 and PI&#61;10. Values of PM&#61;75 and PI&#61;20</p>
</li>
</ol>
<p>give results which are very similar, but often require as much as ten times the processing time   when compared with values of PM&#61;80 and PI&#61;10.</p>
<p><strong>Minscore:</strong> The alignment of a tandem repeat must meet or exceed this alignment score to be  reported. For example, if we set the matching weight to 2 and the minimun score to 50, assuming  perfect alignment, we will need to align at least 25 characters to meet the minimum score  &#40;for example 5 copies with a period of size 5&#41;.</p>
<p><strong>Maxperiod:</strong> Period size is the program&#39;s best guess at the pattern size of the tandem repeat.  The program will find all repeats with period size between 1 and 2000, but the output can be  limited to a smaller range.</p>
<p><strong>-m:</strong> This is an optional parameter and when present instructs the program to generate a masked  sequence file. The masked sequence file is a FASTA format file containing a copy of the sequence   with every location that occurred in a tandem repeat changed to the letter &#39;N&#39;. The word &quot;masked&quot;    is added to the sequence description line just after the &#39;&gt;&#39; character.</p>
<p><strong>-f:</strong> If this option is present, flanking sequence around each repeat is recorded in the alignment  file. This may be useful for PCR primer determination. Flanking sequence consists of the 500   nucleotides on each side of a repeat.</p>
<p><strong>-d:</strong> A data file is produced if this option is present. This file is a text file which contains  the same information, in the same order, as the summary table file, plus consensus pattern and   repeat sequences. This file contains no labeling and is suitable for additional processing, for   example with a perl script, outside of the program.</p>
<p><strong>-h:</strong> suppress HTML output &#40;this automatically switches -d to ON&#41;</p>
<p><strong>-l &lt;n&gt;:</strong> Specifies that the longest TR array expected in the input is at most n million bp long.  The default is 2 &#40;for 2 million&#41;. Setting this option too high may result in an error message if   you did not have enough availablememory. We have only tested this option uo to value 29.</p>
<p><strong>-u:</strong> Prints the help/usage message above</p>
<p><strong>-v:</strong> Prints the version information</p>
<p><strong>-ngs:</strong> More compact .dat output on multisequence files, returns 0 on success. You may pipe input  in with this option using - for file name. Short 50 flanks are appended to .dat output. .dat   output actually goes to stdout instead of file. Sequence headers are displayed in output as   @header. Only headers containing repeats are shown.</p>
</blockquote>
<p>Using recommended parameters the command line will look something like:</p>
<code>trf yoursequence.txt 2 7 7 80 10 50 500 -f -d -m</code>
<p>Once the program starts running it will print update messages to the screen. The word &quot;Done&quot; will be  printed when the program finishes.</p>
<p>For single sequence input files there will be at least two HTML format output files, a repeat table  file and an alignment file. If the number of repeats found is greater than 120, multiple linked  repeat tables are produced. The links to the other tables appear at the top and the bottom of each  table. To view the results start by opening the first repeat table file with your web browser. This  file has the extension &quot;.1.html&quot;. Alignment files can be accessed from the repeat table files.  Alignment files end with the &quot;.txt.html&quot; extension.</p>
<p>For input files containing multiple sequences a summary page is produced that links to the output of  individual sequences. This file has the extension &quot;summary.html&quot;. You should start by opening this  file if your input had multiple sequences in the same file. Also note that the output files of  individual sequences will have an identifier of the form &quot;.sn.&quot; &#40; n an integer&#41; embedded in the name  indicating the index of the sequence in the input file. The identifier is omitted for single sequence  input files. </p>
<h3 id="-d输出结果说明"><a href="#-d输出结果说明" class="header-anchor">-d输出结果说明</a></h3>
<p>加上<code>-d</code>指令后,会输出一个<code>.dat</code>文件,包含所有网页文件中的结果,方便用于perl脚本的操作,输出文件为:</p>
<blockquote>
<p>Sequence: chr1</p>
</blockquote>
<p>Parameters: 2 7 7 80 10 50 2000 1088 1115 2 14.0 2 100 0 56 50 0 50 0 1.00 GA GAGAGAGAGAGAGAGAGAGAGAGAGAGA 8812 8849 2 19.5 2 94 5 69 52 0 0 47 1.00 AT ATATAATATATATATATATATATATATATATATATATA</p>
<p>格式为:</p>
<table><tr><th align="right">Col</th><th align="right">Means</th></tr><tr><td align="right">1</td><td align="right">start</td></tr><tr><td align="right">2</td><td align="right">end</td></tr><tr><td align="right">3</td><td align="right">Period Size</td></tr><tr><td align="right">4</td><td align="right">Copy Number</td></tr><tr><td align="right">5</td><td align="right">Consensus Size</td></tr><tr><td align="right">6</td><td align="right">Percent Matches</td></tr><tr><td align="right">7</td><td align="right">Percent Indels</td></tr><tr><td align="right">8</td><td align="right">Score</td></tr><tr><td align="right">9</td><td align="right">A</td></tr><tr><td align="right">10</td><td align="right">C</td></tr><tr><td align="right">11</td><td align="right">G</td></tr><tr><td align="right">12</td><td align="right">T</td></tr><tr><td align="right">13</td><td align="right">Entropy &#40;0-2&#41;</td></tr><tr><td align="right">14</td><td align="right">Motif seq</td></tr><tr><td align="right">15</td><td align="right">Repeat seq</td></tr></table>


</div>
<!-- article style -->


 <hr>

<script src="https://utteranc.es/client.js"
    repo="songtaogui/blog_comments"
    issue-term="pathname"
    theme="github-light"
    crossorigin="anonymous"
    async>
</script>


<hr> 

<div class="container">
    
    <footer class="site-footer">
        <p class="powered-by">
            Total visits: <span id="busuanzi_value_site_pv"></span>.<br>
            Built with <a href="https://franklinjl.org" target="_blank" rel="noopener">Franklin.jl</a> and <a
                href="https://julialang.org" target="_blank" rel="noopener">Julia</a>.<br>
            The text is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>.<br>
            The code is licensed under the <a href="https://mit-license.org/">MIT License</a>.<br>
            <strong>&copy; Songtao Gui</strong>. Last modified: September 13, 2022.

            <!-- <span class="float-right" aria-hidden="true">
        <a href="#" class="back-to-top">
          <span class="button_icon">
            <i class="fas fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span> -->

        </p>
    </footer>
    
</div>
</div>


</div>
<!--article container -->
</article>


<!-- CONTENT ENDS HERE -->

 <script src="/libs/highlight/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();hljs.configure({tabReplace: '    '});</script>
 
<!-- 
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js
    integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js
    integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js
    integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js
    integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script>
<script src="/libs/academic/academic.min.js"></script>
<script src="/libs/pure/ui.js"></script>
-->


</body>


 <a class="js-back-to-top back-to-top-button" href="#">
    <img src="/assets/img/up.svg" style="width: 100px;">
</a>
<!-- <button class="js-back-to-top back-to-top-button" title="BackToTop">︽</button> -->
<script src="https://cdn.staticfile.org/jquery/2.2.4/jquery.min.js"></script>
<script>
    $(function () {
        var $win = $(window);
        var $backToTop = $('.js-back-to-top');
        // 当用户滚动到离顶部100像素时，展示回到顶部按钮
        $win.scroll(function () {
            if ($win.scrollTop() > 100) {
                $backToTop.show();
            } else {
                $backToTop.hide();
            }
        });
        // 当用户点击按钮时，通过动画效果返回头部
        $backToTop.click(function () {
            $('html, body').animate({
                scrollTop: 0
            }, 200);
        });
    });
</script>
<style>
    /* back to top button */
    .back-to-top-button {
        display: none;
        position: fixed;
        width: 50px;
        bottom: 3%;
        right: 3%;
        z-index: 99;
        font-size: 15px;
        border: 0px solid #696969;
        background-color: #fff;
        color: #222222;
        outline: none;
        cursor: pointer;
        padding: 7px;
        border-radius: 40px;
    }

    .back-to-top-button:hover {
        background-color: #c0dbf5;
        color: #fff;
    }
</style> 


<!-- PJAX配置: 先不搞了! -->
<!--   
  <script src="https://cdn.jsdelivr.net/npm/pjax@0.2.8/pjax.js"></script>
  
  <script>
    var pjax = new Pjax({
      elements: "a",
      selectors: [
        "title",
        "meta[name=description]",
        ".the-header",
        ".the-content",
        ".the-sidebar",
      ]
    })
  </script> -->

</html>


