
<!doctype html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="google-site-verification" content="sKCEBeALSs13uui0cWMviq9hlm0rPByKQKNHnOdq-d0" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="author" content="Songtao Gui">

    
    <meta name="description" content="Brief usage of BUSCO: estimating genome/transcript/protein completeness based on Single Copy Orthologs
">
    

    
    
    <meta name="keywords" content="busco-usage">
    <title>busco-usage</title>
    

    <link rel="icon" href="/assets/favicon.svg">
    <link rel="stylesheet" href="https://cdn.staticfile.org/font-awesome/4.7.0/css/font-awesome.css">
    <link rel=preconnect href="https://fonts.gstatic.com" crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.1/css/all.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css"
    crossorigin=anonymous>
<link rel=stylesheet href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css" crossorigin=anonymous>
<link rel=stylesheet
    href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap"
    crossorigin=anonymous>

<!-- locally served -->
<link rel="stylesheet" href="/css/pure.css">
<link rel=stylesheet href="/libs/academic/academic.min.css">
<link rel="stylesheet" href="/css/layout.css">
<link rel="stylesheet" href="/css/extra.css">
<link rel=stylesheet href="/css/codeset.css">
<link rel=stylesheet href="/css/admonition.css">
<link rel=stylesheet href="/css/cv.css">

<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>

<!-- nutshell + highlight + katex + plotly -->

<link rel=stylesheet href="/libs/highlight/styles/atom-one-light.min.css">



    <script src="https://cdn.jsdelivr.net/npm/@svgdotjs/svg.js@3.0/dist/svg.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js"
        integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script>
</head>

<body id=top data-spy="scroll" data-offset="70" data-target=#navbar-main>

     <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main>
    <div class=container>
        <!-- Main name -->
        <div class="d-none d-lg-inline-flex">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Button for narrow mode -->
        <button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar
            aria-expanded=false aria-label="Toggle navigation">
            <span><i class="fas fa-bars"></i></span>
        </button>
        <!-- Main name for mobile mode -->
        <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
            <a class=navbar-brand href="/">Songtao Gui</a>
        </div>
        <!-- Menu items -->
        <div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content>
            <ul class="navbar-nav d-md-inline-flex">
                <li class=nav-item><a class=nav-link href="/posts/"><span>Posts</span></a></li>
                <!-- <li class=nav-item><a class=nav-link href="/notebooks/"><span>Notebooks</span></a></li> -->
                <li class=nav-item><a class=nav-link href="/courses/"><span>Courses</span></a></li>
                <li class=nav-item><a class=nav-link href="/achievements/"><span>Achievements</span></a></li>
                <li class=nav-item><a class=nav-link href="/cv/"><span>CurriculumVitae</span></a></li>
                <li class=nav-item><a class=nav-link href="/tag/"><span>Tags</span></a></li>
                <!-- <li class=nav-item><a class=nav-link href="/giants/"><span>Giants</span></a></li> -->


                <!-- 
        <li class=nav-item><a class=nav-link ><span> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></a></li>
        <li class=nav-item>
          <a class=nav-link>
            <span>
              <script src="/libs/lunr/lunr.js"></script>
              <script src="lunr.stemmer.support.js"></script>
              <script src="/libs/lunr/lunr.zh.js"></script>
              <script src="/libs/lunr/lunr_index.js"></script>
              <script src="/libs/lunr/lunrclient.min.js"></script>
              <form id="lunrSearchForm" name="lunrSearchForm">
                <input class="search-input" name="q" placeholder="Enter search term" type="text">
                <input type="submit" value="Search" formaction="/search/index.html">
              </form>
            </span>
          </a>
        </li>
        -->

            </ul>
        </div>
        <div class="nav-icon d-none d-lg-inline-flex">
            <a class=nav-linksmall href="/notebooks" target="_blank"><span><img class="nav-img"
                        src="/assets/img/notebook.svg"></span></a>
            <a class=nav-linksmall href="/giants" target="_blank"><span><img class="nav-img"
                        src="/assets/img/giant.svg"></span></a>
            <a class=nav-linksmall href="/feed.xml" target="_blank"><span><img class="nav-img"
                        src="/assets/img/rss.svg"></span></a>
        </div>
        <div class="nav-icon ">
            <a class=nav-linksmall href="/musics" target="_blank"><span><img class="nav-img"
                        src="/assets/img/music.svg"></span></a>
        </div>

    </div>
</nav> 

    <span class="js-widget-page d-none"></span>

    
    <article class=article>
        <div class="article-container pt-3">
             <h1>busco-usage</h1> 
        </div>
        
        <div class="article-container">
            <!-- 关闭article-style, 不然跟nutshell不兼容 -->
            <!-- <div class="gst-style"> -->
            <div class="layout-main">
                <!-- <div class="article-style"> -->
                
                
                <div class="franklin-content"><p> <div class="tags"><a href="/tag/" id="tag-icon"><svg width="20" height="20" viewBox="0 0 512 512"><defs><style>.cls-1{fill:#141f38}</style></defs><path class="cls-1" d="M215.8 512a76.1 76.1 0 0 1-54.17-22.44L22.44 350.37a76.59 76.59 0 0 1 0-108.32L242 22.44A76.11 76.11 0 0 1 296.2 0h139.2A76.69 76.69 0 0 1 512 76.6v139.19A76.08 76.08 0 0 1 489.56 270L270 489.56A76.09 76.09 0 0 1 215.8 512zm80.4-486.4a50.69 50.69 0 0 0-36.06 14.94l-219.6 219.6a51 51 0 0 0 0 72.13l139.19 139.19a51 51 0 0 0 72.13 0l219.6-219.61a50.67 50.67 0 0 0 14.94-36.06V76.6a51.06 51.06 0 0 0-51-51zm126.44 102.08A38.32 38.32 0 1 1 461 89.36a38.37 38.37 0 0 1-38.36 38.32zm0-51a12.72 12.72 0 1 0 12.72 12.72 12.73 12.73 0 0 0-12.72-12.76z"/><path class="cls-1" d="M217.56 422.4a44.61 44.61 0 0 1-31.76-13.16l-83-83a45 45 0 0 1 0-63.52L211.49 154a44.91 44.91 0 0 1 63.51 0l83 83a45 45 0 0 1 0 63.52L249.31 409.24a44.59 44.59 0 0 1-31.75 13.16zm-96.7-141.61a19.34 19.34 0 0 0 0 27.32l83 83a19.77 19.77 0 0 0 27.31 0l108.77-108.7a19.34 19.34 0 0 0 0-27.32l-83-83a19.77 19.77 0 0 0-27.31 0l-108.77 108.7z"/><path class="cls-1" d="M294.4 281.6a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85zM256 320a12.75 12.75 0 0 1-9.05-3.75l-51.2-51.2a12.8 12.8 0 0 1 18.1-18.1l51.2 51.2A12.8 12.8 0 0 1 256 320zM217.6 358.4a12.75 12.75 0 0 1-9-3.75l-51.2-51.2a12.8 12.8 0 1 1 18.1-18.1l51.2 51.2a12.8 12.8 0 0 1-9.05 21.85z"/></svg></a><a href="/tag/bioinformatics/">bioinformatics</a>, <a href="/tag/genomics/">genomics</a>&nbsp;&nbsp;&nbsp;&nbsp;<a id="tag-icon"><svg width="20" height="20" class="icon" viewBox="0 0 1024 1024" version="1.1" p-id="3825" width="64" height="64"><path d="M512 416a96 96 0 1 0 0 192 96 96 0 0 0 0-192z m511.952 102.064c-0.016-0.448-0.064-0.864-0.096-1.296a8.16 8.16 0 0 0-0.08-0.656c0-0.32-0.064-0.624-0.128-0.928-0.032-0.368-0.064-0.736-0.128-1.088-0.032-0.048-0.032-0.096-0.032-0.144a39.488 39.488 0 0 0-10.704-21.536c-32.672-39.616-71.536-74.88-111.04-107.072-85.088-69.392-182.432-127.424-289.856-150.8-62.112-13.504-124.576-14.064-187.008-2.64-56.784 10.384-111.504 32-162.72 58.784-80.176 41.92-153.392 99.696-217.184 164.48-11.808 11.984-23.552 24.224-34.288 37.248-14.288 17.328-14.288 37.872 0 55.216 32.672 39.616 71.52 74.848 111.04 107.056 85.12 69.392 182.448 127.408 289.888 150.784 62.096 13.504 124.608 14.096 187.008 2.656 56.768-10.4 111.488-32 162.736-58.768 80.176-41.936 153.376-99.696 217.184-164.48 11.792-12 23.536-24.224 34.288-37.248 5.712-5.872 9.456-13.44 10.704-21.568l0.032-0.128a12.592 12.592 0 0 0 0.128-1.088c0.064-0.304 0.096-0.624 0.128-0.928l0.08-0.656 0.096-1.28c0.032-0.656 0.048-1.296 0.048-1.952l-0.096-1.968zM512 704c-106.032 0-192-85.952-192-192s85.952-192 192-192 192 85.968 192 192c0 106.048-85.968 192-192 192z" p-id="3826"></path></svg></a><span id="busuanzi_value_page_pv"></span></div></p>
<h2 id="busco_安装"><a href="#busco_安装" class="header-anchor">BUSCO 安装</a></h2>
<ol>
<li><p>BUSCO使用需要依赖软件: </p>
<ul>
<li><p>python3; </p>
</li>
<li><p>hmmer; </p>
</li>
<li><p>NCBI BLAST&#43;; </p>
</li>
<li><p>Augustus 3.0.x &#40;genome only&#41;; </p>
<ul>
<li><p>boost</p>
</li>
<li><p>zlib</p>
</li>
<li><p>bamtools &#40;编译需要cmake&#41;</p>
</li>
</ul>
</li>
<li><p>EMBOSS tools 6.x.x</p>
</li>
</ul>
</li>
</ol>
<p>安装相关软件并设置环境变量：</p>
<h3 id="python3"><a href="#python3" class="header-anchor">python3 </a></h3>
<pre><code class="language-shell">wget https://www.python.org/ftp/python/3.5.2/Python-3.5.2.tgz
tar zxvf Python-3.5.2.tgz
cd Python-3.5.2
./configure --prefix&#61;&#36;SFTW/app/python35
make -j8 &amp;&amp; make install
echo &#39;export PATH&#61;SSFTW/app/python35/bin:&#36;PATH&#39; &gt;&gt; ~/.zshrc &amp;&amp; source ~/.zshrc #addpath</code></pre>
<h3 id="hmmer"><a href="#hmmer" class="header-anchor">hmmer:</a></h3>
<pre><code class="language-shell">wget http://eddylab.org/software/hmmer3/3.1b2/hmmer-3.1b2.tar.gz
tar zxvf hmmer-3.1b2.tar.gz
cd hmmer-3.1b2
./configure --prefix&#61;&#36;SFTW/app/hmmer
make -j8 &amp;&amp; make install
addpath</code></pre>
<h3 id="boost"><a href="#boost" class="header-anchor">boost：</a></h3>
<pre><code class="language-shell">wget https://sourceforge.net/projects/boost/files/boost/1.61.0/boost_1_61_0.tar.gz
tar zxvf boost_1_61_0.tar.gz
cd boost_1_61_0
./bootstrap.sh --prefix&#61;&#36;SFTW/app/boost
./b2
./bjam install
添加环境变量到 ~/.zshrc
export BOOST_INCLUDE&#61;&#36;SFTW/app/boost/include
export BOOST_LIB&#61;&#36;SFTW/app/boost/lib</code></pre>
<h3 id="zlib"><a href="#zlib" class="header-anchor">zlib:</a></h3>
<pre><code class="language-shell">wget http://zlib.net/zlib-1.2.8.tar.gz
tar zxvf zlib-1.2.8.tar.gz
cd zlib-1.2.8
./configure --prefix&#61;/lustre/home/jiangff/software/app/zlib
make -j8 &amp;&amp; make install
cp -r /lustre/home/jiangff/software/app/zlib/* &#36;SFTW 
#&#36;sftw主目录下的include 和lib已经添加环境变量 所以直接复制过去</code></pre>
<h3 id="cmake"><a href="#cmake" class="header-anchor">cmake：</a></h3>
<pre><code class="language-shell">wget https://cmake.org/files/v3.6/cmake-3.6.1.tar.gz --no-check-certificate
tar zxvf |cd 
./configure --prefix&#61;&#36;SFTW/app/cmake
make -j8 &amp;&amp; make install
cd bin|addpath</code></pre>
<h3 id="bamtools"><a href="#bamtools" class="header-anchor">bamtools：</a></h3>
<pre><code class="language-shell">wget https://codeload.github.com/pezmaster31/bamtools/zip/master
mv master bamtools.zip
unzip bamtools.zip
cd bamtools-master
mkdir build
cd build
cmake ..
make -j8
cd ..
cp -r bin/ include/ lib/ &#36;SFTW</code></pre>
<h3 id="augustus"><a href="#augustus" class="header-anchor">Augustus:</a></h3>
<pre><code class="language-shell">wget http://bioinf.uni-greifswald.de/augustus/binaries/augustus.current.tar.gz
tar | cd
make -j8
cd ..
mv augustus-3.2.2/ /lustre/home/jiangff/software/app/augustus
addpath: &#36;SFTW/app/augustus/bin | scripts
#在~/.zshrc 加上：
export AUGUSTUS_CONFIG_PATH&#61;&#36;SFTW/app/augustus/config #不加这个不行！</code></pre>
<h3 id="emboss_tools"><a href="#emboss_tools" class="header-anchor">EMBOSS tools:</a></h3>
<pre><code class="language-shell">wget http://debian.rub.de/ubuntu/pool/universe/e/emboss/emboss_6.6.0.orig.tar.gz
tar |cd 
./configure --prefix...
make &amp;&amp; make install
addpath</code></pre>
<h3 id="busco"><a href="#busco" class="header-anchor">BUSCO</a></h3>
<pre><code class="language-shell">wget http://busco.ezlab.org/files/BUSCO_v1.2.tar.gz
tar zxvf BUSCO_v1.2.tar.gz
#解压即可用
mv to &#36;SFTW/app/BUSCO
#在~/.zshrc添加：
alias BUSCO&#61;&quot;python /lustre/home/jiangff/software/app/BUSCO/BUSCO_v1.2.py&quot;

#下载资料库（以真核资料库为例）：
    wget http://busco.ezlab.org/files/eukaryota_buscos.tar.gz
    tar zxvf eukaryota_buscos.tar.gz
    mv eukaryota &#36;SFTW/app/BUSCO</code></pre>
<h2 id="busco使用"><a href="#busco使用" class="header-anchor">BUSCO使用</a></h2>
<h3 id="busco_quick_start"><a href="#busco_quick_start" class="header-anchor">BUSCO quick start</a></h3>
<ul>
<li><p>Genome assembly assessment:</p>
<p><code>shell   python BUSCO_v1.1b.py -o NAME -in ASSEMBLY -l LINEAGE –m genome</code></p>
</li>
</ul>
<p><strong>NAME</strong> name to use for the run and all temporary files <strong>ASSEMBLY</strong> genome assembly file in fasta format <strong>LINEAGE</strong> path to the BUSCO lineage data &#40;i.e. /path/to/lineage&#41;</p>
<ul>
<li><p>­Gene set assessment:</p>
<p><code>shell   python BUSCO_v1.1b.py -o NAME -in GENE_SET -l LINEAGE -m OGS</code></p>
</li>
<li><p>­Transcriptome assessment:</p>
</li>
</ul>
<p>&#96;&#96;&#96;
python BUSCO_v1.1b.py -o NAME -in TRANSCRIPTOME -l LINEAGE -m trans
&#96;&#96;&#96;</p>
<h3 id="busco使用__2"><a href="#busco使用__2" class="header-anchor">BUSCO使用</a></h3>
<h4 id="genome_assembly_assessment"><a href="#genome_assembly_assessment" class="header-anchor"><ol>
<li><p>Genome assembly assessment:</p>
</li>
</ol>
</a></h4>
<pre><code class="language-sh">python BUSCO_v1.1b.py -o NAME -in ASSEMBLY -l LINEAGE –m genome

#NAME      name to use for the run and all temporary files
#ASSEMBLY  genome assembly file in fasta format
#LINEAGE   path to the BUSCO lineage data</code></pre>
<h4 id="ol_start2_gene_set_assessment"><a href="#ol_start2_gene_set_assessment" class="header-anchor"><ol start="2">
<li><p>Gene set assessment:</p>
</li>
</ol>
</a></h4>
<pre><code class="language-sh">python BUSCO_v1.1b.py -o NAME -in GENE_SET -l LINEAGE -m OGS

#NAME       name to use for the run and temporary files
#GENE_SET   gene set protein sequence file in fasta format
#Lineage    path to the BUSCO lineage data &#40;i.e. /path/to/lineage&#41;</code></pre>
<h4 id="ol_start3_transcriptome_assessment"><a href="#ol_start3_transcriptome_assessment" class="header-anchor"><ol start="3">
<li><p>Transcriptome assessment:</p>
</li>
</ol>
</a></h4>
<pre><code class="language-sh">python BUSCO_v1.1b.py -o NAME -in TRANSCRIPTOME -l LINEAGE -m trans

#NAME           name to use for the run and temporary files
#TRANSCRIPTOME  transcript set sequence file in fasta format
#LINEAGE        path to the BUSCO lineage data &#40;i.e. /path/to/lineage&#41;</code></pre>
<h3 id="busco_options"><a href="#busco_options" class="header-anchor">BUSCO options</a></h3>
<pre><code class="language-sh">python BUSCO_v1.1b.py -in INPUT -o OUTPUT -l LINEAGE -m MODE</code></pre>
<p><strong>&#91;Options&#93;</strong></p>
<p>1-­ Mandatory arguments</p>
<p>-o name Name used for naming output files
-in input_file Genome assembly / gene set / transcriptome in fasta format
-l lineage Path to the BUSCO lineage data to be used
Example: -­l /path/to/metazoa
-m mode Mode of analysis
Valid options: genome, ogs, trans
Default: genome</p>
<p>2-­ Optional arguments</p>
<p>-h –help Print help
-c integer Number of CPU threads to be used
Default: 1
-sp species Select from the pre-­computed Augustus metaparameters
Selecting a closely-­related species usually produces better results
Valid options: see Augustus help for list of options
Default: generic
-e evalue Use a custom blast e-­value cutoff
Default: 0.01
-f Force overwriting of results files from a previous run with the same name
--flank N Custom flanking genomic regions in base pairs &#40;bp&#41;
Used when extending selected candidate regions before gene prediction
Default: Automatically calculated flank sizes based on genome size
-­-­long Performs full optimization for Augustus gene finding training
Default: Off</p>
<h3 id="busco_output"><a href="#busco_output" class="header-anchor">BUSCO Output</a></h3>
<p>Successful execution of the BUSCO assessment pipeline will create a directory named name_OUTPUT where ‘name’ is your assigned name for the assessment run. The directory will contain several files and directories:</p>
<h4 id="files"><a href="#files" class="header-anchor">Files</a></h4>
<p>short_summary_    Contains summary results in BUSCO notation and a brief breakdown of the metrics

full_table_       Complete results in tabular format with coordinates, scores and lengths of 
                  BUSCO matches

training_set_     Set of complete BUSCO matches used for training Augustus 
                  Only created during genome assessment

_tblastn          Results in tabular format of tBLASTn searches with BUSCO consensus sequences</p>
<h4 id="directories"><a href="#directories" class="header-anchor">Directories</a></h4>
<p>augustus_         Augustus-­predicted genes
                  Only created during genome assessment
augutus_proteins  Corresponding Augustus-­predicted proteins
                  Only created during genome assessment
selected          Complete BUSCO matches, used for training Augustus
gb                Complete BUSCO matches, GenBank format
gffs              Complete BUSCO matches, GFF format
hmmer_output      Tabular format HMMER output of searches with BUSCO HMMs</p>
<h3 id="例子"><a href="#例子" class="header-anchor">例子</a></h3>
<p>BUSCO 评价基因组组装质量</p>
<p>Ask for Plant Lineage data: <code>felipe.simao@unige.ch</code> NO ANSWER</p>
<p>Using Eukaryote instead:</p>
<pre><code class="language-shell">#BUSCO: aliased to &quot;python XXX/XXX/BUSCO/BUSCO_v1.2.py&quot;

BUSCO -o Rad -in AAA.fasta -l &#36;SFTW/app/BUSCO/eukaryota -m genome</code></pre>
<p>结果：</p>
<pre><code class="language-julia">Summarized benchmarks in BUSCO notation:
        C:0&#37;&#91;D:0&#37;&#93;,F:0&#37;,M:0&#37;,n:429

        190     Complete BUSCOs
        139     Complete and single-copy BUSCOs
        51      Complete and duplicated BUSCOs
        32      Fragmented BUSCOs
        207     Missing BUSCOs
        429     Total BUSCO groups searched</code></pre>


</div>
<!-- article style -->


 <hr>

<script src="https://utteranc.es/client.js"
    repo="songtaogui/blog_comments"
    issue-term="pathname"
    theme="github-light"
    crossorigin="anonymous"
    async>
</script>


<hr> 

<div class="container">
    
    <footer class="site-footer">
        <p class="powered-by">
            Total visits: <span id="busuanzi_value_site_pv"></span>.<br>
            Built with <a href="https://franklinjl.org" target="_blank" rel="noopener">Franklin.jl</a> and <a
                href="https://julialang.org" target="_blank" rel="noopener">Julia</a>.<br>
            The text is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>.<br>
            The code is licensed under the <a href="https://mit-license.org/">MIT License</a>.<br>
            <strong>&copy; Songtao Gui</strong>. Last modified: September 13, 2022.

            <!-- <span class="float-right" aria-hidden="true">
        <a href="#" class="back-to-top">
          <span class="button_icon">
            <i class="fas fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span> -->

        </p>
    </footer>
    
</div>
</div>


</div>
<!--article container -->
</article>


<!-- CONTENT ENDS HERE -->

 <script src="/libs/highlight/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();hljs.configure({tabReplace: '    '});</script>
 
<!-- 
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js
    integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js
    integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js
    integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js
    integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script>
<script src="/libs/academic/academic.min.js"></script>
<script src="/libs/pure/ui.js"></script>
-->


</body>


 <a class="js-back-to-top back-to-top-button" href="#">
    <img src="/assets/img/up.svg" style="width: 100px;">
</a>
<!-- <button class="js-back-to-top back-to-top-button" title="BackToTop">︽</button> -->
<script src="https://cdn.staticfile.org/jquery/2.2.4/jquery.min.js"></script>
<script>
    $(function () {
        var $win = $(window);
        var $backToTop = $('.js-back-to-top');
        // 当用户滚动到离顶部100像素时，展示回到顶部按钮
        $win.scroll(function () {
            if ($win.scrollTop() > 100) {
                $backToTop.show();
            } else {
                $backToTop.hide();
            }
        });
        // 当用户点击按钮时，通过动画效果返回头部
        $backToTop.click(function () {
            $('html, body').animate({
                scrollTop: 0
            }, 200);
        });
    });
</script>
<style>
    /* back to top button */
    .back-to-top-button {
        display: none;
        position: fixed;
        width: 50px;
        bottom: 3%;
        right: 3%;
        z-index: 99;
        font-size: 15px;
        border: 0px solid #696969;
        background-color: #fff;
        color: #222222;
        outline: none;
        cursor: pointer;
        padding: 7px;
        border-radius: 40px;
    }

    .back-to-top-button:hover {
        background-color: #c0dbf5;
        color: #fff;
    }
</style> 


<!-- PJAX配置: 先不搞了! -->
<!--   
  <script src="https://cdn.jsdelivr.net/npm/pjax@0.2.8/pjax.js"></script>
  
  <script>
    var pjax = new Pjax({
      elements: "a",
      selectors: [
        "title",
        "meta[name=description]",
        ".the-header",
        ".the-content",
        ".the-sidebar",
      ]
    })
  </script> -->

</html>


